{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# This script provides some reporting on the resolver data from identifiers.org\n",
    "#\n",
    "# Author: Manuel Bernal Llinares <mbdebian@gmail.com>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "import time\n",
    "import random\n",
    "import pandas\n",
    "import requests\n",
    "import threading\n",
    "import numpy as np\n",
    "import multiprocessing\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter\n",
    "from multiprocessing.pool import ThreadPool\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Endpoint from where the information is coming\n",
    "identifiersorg_resolver_data_url = \"https://identifiers.org/rest/collections/expand\"\n",
    "metadata_service_endpoint_from_url = \"http://localhost:8082/getMetadataForUrl\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize pseudo-random number generator\n",
    "random.seed(time.time())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Helpers\n",
    "def make_rest_request_content_type_json(url):\n",
    "    # TODO - Magic number here!!!\n",
    "    n_attempts = 42\n",
    "    response = None\n",
    "    while n_attempts:\n",
    "        n_attempts -= 1\n",
    "        try:\n",
    "            response = requests.get(url, headers={\"Content-Type\": \"application/json\"})\n",
    "        except Exception as e:\n",
    "            # Any possible exception counts towards the attempt counter\n",
    "            # Random wait - TODO - Another magic number!!!\n",
    "            time.sleep(random.randint(30))\n",
    "            continue\n",
    "        if response.ok:\n",
    "            return response.json()\n",
    "        # Random wait - TODO - Another magic number!!!\n",
    "        time.sleep(random.randint(10))\n",
    "    response.raise_for_status()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_metadata_for_url(url):\n",
    "    \"\"\"\n",
    "    This Helper queries the metadata service with a URL, \n",
    "    and returns its response back to the caller for further interpretation\n",
    "    \"\"\"\n",
    "    n_attempts = 42\n",
    "    response = None\n",
    "    while n_attempts:\n",
    "        n_attempts -= 1\n",
    "        try:\n",
    "            response = requests.post(metadata_service_endpoint_from_url, json={\"url\": url})\n",
    "        except Exception as e:\n",
    "            # Any possible exception counts towards the attempt counter\n",
    "            # Random wait - TODO - Another magic number!!!\n",
    "            time.sleep(random.randint(3))\n",
    "            continue\n",
    "        if response.ok:\n",
    "            print(\"[METADATA][OK] - '{}'\".format(url))\n",
    "            break\n",
    "        else:\n",
    "            print(\"[METADATA][ERROR] - '{}'\".format(url))\n",
    "            break\n",
    "        # Random wait - TODO - Another magic number!!!\n",
    "    return response"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the resolver data\n",
    "resolver_dump = make_rest_request_content_type_json(identifiersorg_resolver_data_url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Workout how many prefixes there are in identifiers.org\n",
    "prefixes = [pid_entry['prefix'] for pid_entry in resolver_dump]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chebi',\n",
       " 'ensembl',\n",
       " 'ec-code',\n",
       " 'uniprot',\n",
       " 'taxonomy',\n",
       " 'biomodels.db',\n",
       " 'miriam.collection',\n",
       " 'icd',\n",
       " 'intact',\n",
       " 'interpro',\n",
       " 'kegg.pathway',\n",
       " 'kegg.compound',\n",
       " 'kegg.reaction',\n",
       " 'pubmed',\n",
       " 'omim',\n",
       " 'pirsf',\n",
       " 'reactome',\n",
       " 'doi',\n",
       " 'pdb',\n",
       " 'go',\n",
       " 'sgd',\n",
       " 'sbo',\n",
       " 'kegg.drug',\n",
       " 'kegg.glycan',\n",
       " 'wormbase',\n",
       " 'pfam',\n",
       " 'insdc',\n",
       " 'flybase',\n",
       " 'wormpep',\n",
       " 'prosite',\n",
       " 'pubchem.substance',\n",
       " 'pubchem.compound',\n",
       " 'arxiv',\n",
       " 'arrayexpress',\n",
       " 'mgi',\n",
       " 'sabiork.reaction',\n",
       " 'refseq',\n",
       " 'tcdb',\n",
       " 'uniparc',\n",
       " 'mint',\n",
       " 'dip',\n",
       " 'signaling-gateway',\n",
       " 'resid',\n",
       " 'rgd',\n",
       " 'tair.protein',\n",
       " 'tair.gene',\n",
       " 'tair.locus',\n",
       " 'hmdb',\n",
       " 'lipidmaps',\n",
       " 'peptideatlas',\n",
       " 'geo',\n",
       " 'eco',\n",
       " 'mod',\n",
       " 'sgd.pathways',\n",
       " 'biogrid',\n",
       " 'merops',\n",
       " 'panther.family',\n",
       " 'prints',\n",
       " 'ligandexpo',\n",
       " 'aclame',\n",
       " 'isbn',\n",
       " 'pride',\n",
       " '3dmet',\n",
       " 'fma',\n",
       " 'matrixdb.association',\n",
       " 'ncbigene',\n",
       " 'kegg.genes',\n",
       " 'brenda',\n",
       " 'pubchem.bioassay',\n",
       " 'pathwaycommons',\n",
       " 'hovergen',\n",
       " 'mmmp:biomaps',\n",
       " 'wikipathways',\n",
       " 'macie',\n",
       " 'mirbase',\n",
       " 'zfin',\n",
       " 'hgnc',\n",
       " 'so',\n",
       " 'rhea',\n",
       " 'unipathway',\n",
       " 'chembl.compound',\n",
       " 'chembl.target',\n",
       " 'sabiork.kineticrecord',\n",
       " 'lgic',\n",
       " 'atc',\n",
       " 'pharmgkb.pathways',\n",
       " 'pharmgkb.disease',\n",
       " 'pharmgkb.drug',\n",
       " 'ttd.drug',\n",
       " 'ttd.target',\n",
       " 'neurondb',\n",
       " 'neuromorpho',\n",
       " 'chemidplus',\n",
       " 'biosystems',\n",
       " 'ctd.chemical',\n",
       " 'ctd.disease',\n",
       " 'ctd.gene',\n",
       " 'bionumbers',\n",
       " 'drugbank',\n",
       " 't3db',\n",
       " 'smpdb',\n",
       " 'phosphosite.protein',\n",
       " 'genedb',\n",
       " 'biomodels.teddy',\n",
       " 'biomodels.kisao',\n",
       " 'psimi',\n",
       " 'cl',\n",
       " 'bto',\n",
       " 'pato',\n",
       " 'pdb-ccd',\n",
       " 'glycomedb',\n",
       " 'lipidbank',\n",
       " 'kegg.orthology',\n",
       " 'prodom',\n",
       " 'smart',\n",
       " 'cdd',\n",
       " 'ro',\n",
       " 'mmdb',\n",
       " 'imex',\n",
       " 'irefweb',\n",
       " 'mpid',\n",
       " 'phosphosite.residue',\n",
       " 'neurolex',\n",
       " 'obi',\n",
       " 'sabiork.ec',\n",
       " 'opb',\n",
       " 'jws',\n",
       " 'modeldb',\n",
       " 'subtiwiki',\n",
       " 'pid.pathway',\n",
       " 'doqcs.model',\n",
       " 'doqcs.pathway',\n",
       " 'uo',\n",
       " 'clinicaltrials',\n",
       " 'chemspider',\n",
       " 'ncit',\n",
       " 'biocatalogue.service',\n",
       " 'pr',\n",
       " 'omia',\n",
       " 'chembank',\n",
       " 'csa',\n",
       " 'cgd',\n",
       " 'antweb',\n",
       " 'pmc',\n",
       " 'amoebadb',\n",
       " 'cryptodb',\n",
       " 'plasmodb',\n",
       " 'giardiadb',\n",
       " 'microsporidia',\n",
       " 'toxoplasma',\n",
       " 'trichdb',\n",
       " 'tritrypdb',\n",
       " 'bdgp.insertion',\n",
       " 'beetlebase',\n",
       " 'bold.taxonomy',\n",
       " 'dbest',\n",
       " 'dbprobe',\n",
       " 'dbsnp',\n",
       " 'unists',\n",
       " 'ecogene',\n",
       " 'gabi',\n",
       " 'greengenes',\n",
       " 'grin.taxonomy',\n",
       " 'hinv.locus',\n",
       " 'hinv.transcript',\n",
       " 'hinv.protein',\n",
       " 'homd.seq',\n",
       " 'homd.taxon',\n",
       " 'ird.segment',\n",
       " 'isfinder',\n",
       " 'jcm',\n",
       " 'img.taxon',\n",
       " 'img.gene',\n",
       " 'maizegdb.locus',\n",
       " 'mycobank',\n",
       " 'nbrc',\n",
       " 'pseudomonas',\n",
       " 'gramene.protein',\n",
       " 'gramene.gene',\n",
       " 'gramene.taxonomy',\n",
       " 'gramene.qtl',\n",
       " 'sgn',\n",
       " 'xenbase',\n",
       " 'bioportal',\n",
       " 'miriam.resource',\n",
       " 'edam',\n",
       " 'pmdb',\n",
       " 'arachnoserver',\n",
       " 'biocyc',\n",
       " 'cazy',\n",
       " 'goa',\n",
       " 'paleodb',\n",
       " 'compulyeast',\n",
       " 'disprot',\n",
       " 'echobase',\n",
       " 'eggnog',\n",
       " 'ensembl.bacteria',\n",
       " 'ensembl.protist',\n",
       " 'ensembl.metazoa',\n",
       " 'ensembl.plant',\n",
       " 'ensembl.fungi',\n",
       " 'hcvdb',\n",
       " 'genatlas',\n",
       " 'cath.superfamily',\n",
       " 'cath.domain',\n",
       " 'genefarm',\n",
       " 'gpcrdb',\n",
       " 'hogenom',\n",
       " 'genetree',\n",
       " 'hssp',\n",
       " 'myco.tuber',\n",
       " 'myco.lepra',\n",
       " 'myco.marinum',\n",
       " 'myco.smeg',\n",
       " 'orphanet',\n",
       " 'orthodb',\n",
       " 'peroxibase',\n",
       " 'phylomedb',\n",
       " 'pmap.substratedb',\n",
       " 'pmap.cutdb',\n",
       " 'protclustdb',\n",
       " 'pmp',\n",
       " 'protonet.proteincard',\n",
       " 'protonet.cluster',\n",
       " 'rebase',\n",
       " 'swiss-model',\n",
       " 'vectorbase',\n",
       " 'doid',\n",
       " 'cco',\n",
       " 'mirbase.mature',\n",
       " 'nextprot',\n",
       " 'cas',\n",
       " 'kegg.genome',\n",
       " 'kegg.metagenome',\n",
       " 'narcis',\n",
       " 'jcsd',\n",
       " 'pw',\n",
       " 'insdc.sra',\n",
       " 'scretf',\n",
       " 'pharmgkb.gene',\n",
       " 'mirnest',\n",
       " 'napp',\n",
       " 'noncodev3',\n",
       " 'virsirna',\n",
       " 'elm',\n",
       " 'mimodb',\n",
       " 'sitex',\n",
       " 'bykdb',\n",
       " 'conoserver',\n",
       " 'topfind',\n",
       " 'mipmod',\n",
       " 'cellimage',\n",
       " 'combine.specifications',\n",
       " 'zfin.phenotype',\n",
       " 'zfin.expression',\n",
       " 'cabri',\n",
       " 'cygd',\n",
       " 'huge',\n",
       " 'bindingDB',\n",
       " 'string',\n",
       " 'stitch',\n",
       " 'atcvet',\n",
       " 'phenolexplorer',\n",
       " 'snomedct',\n",
       " 'mesh.2012',\n",
       " 'knapsack',\n",
       " 'cdpd',\n",
       " 'massbank',\n",
       " 'gmd',\n",
       " 'homologene',\n",
       " 'umbbd.compound',\n",
       " 'abs',\n",
       " 'apd',\n",
       " 'chemdb',\n",
       " 'dpv',\n",
       " 'iuphar.receptor',\n",
       " 'aceview.worm',\n",
       " 'asap',\n",
       " 'atcc',\n",
       " 'bdgp.est',\n",
       " 'dictybase.gene',\n",
       " 'imgt.ligm',\n",
       " 'worfdb',\n",
       " 'nextdb',\n",
       " 'soybase',\n",
       " 'hamap',\n",
       " 'rouge',\n",
       " 'arrayexpress.platform',\n",
       " 'cgsc',\n",
       " 'dragondb.dna',\n",
       " 'dragondb.protein',\n",
       " 'dragondb.locus',\n",
       " 'dragondb.allele',\n",
       " 'issn',\n",
       " 'merops.family',\n",
       " 'mo',\n",
       " 'nasc',\n",
       " 'niaest',\n",
       " 'pazar',\n",
       " 'po',\n",
       " 'rnamods',\n",
       " 'treebase',\n",
       " 'tgd',\n",
       " 'tigrfam',\n",
       " 'atfdb.family',\n",
       " 'iuphar.family',\n",
       " 'dbg2introns',\n",
       " 'sdbs',\n",
       " 'vbase2',\n",
       " 'spike.map',\n",
       " 'metlin',\n",
       " 'genecards',\n",
       " 'mmrrc',\n",
       " 'umbbd.reaction',\n",
       " 'umbbd.enzyme',\n",
       " 'umbbd.pathway',\n",
       " 'umbbd.rule',\n",
       " 'mirex',\n",
       " 'dictybase.est',\n",
       " 'imgt.hla',\n",
       " 'flystock',\n",
       " 'opm',\n",
       " 'allergome',\n",
       " 'pombase',\n",
       " 'hpa',\n",
       " 'jaxmice',\n",
       " 'ubio.namebank',\n",
       " 'yetfasco',\n",
       " 'tarbase',\n",
       " 'charprot',\n",
       " 'oma.protein',\n",
       " 'oma.grp',\n",
       " 'ncbiprotein',\n",
       " 'genpept',\n",
       " 'unigene',\n",
       " 'bitterdb.rec',\n",
       " 'bitterdb.cpd',\n",
       " 'bioproject',\n",
       " 'biosample',\n",
       " 'piroplasma',\n",
       " 'unite',\n",
       " 'ncim',\n",
       " 'proglyc',\n",
       " 'polbase',\n",
       " 'nuclearbd',\n",
       " 'supfam',\n",
       " 'ricegap',\n",
       " 'pina',\n",
       " 'tissuelist',\n",
       " 'bacmap.biog',\n",
       " 'hgnc.symbol',\n",
       " 'panther.pathway',\n",
       " 'fairsharing',\n",
       " 'fungidb',\n",
       " 'darc',\n",
       " 'drsc',\n",
       " 'oridb.schizo',\n",
       " 'oridb.sacch',\n",
       " 'pscdb',\n",
       " 'scop',\n",
       " 'ena.embl',\n",
       " 'dommino',\n",
       " 'panther.node',\n",
       " 'ccds',\n",
       " 'lrg',\n",
       " 'hprd',\n",
       " 'gxa.gene',\n",
       " 'gxa.expt',\n",
       " 'metabolights',\n",
       " 'nbn',\n",
       " 'orcid',\n",
       " 'inchi',\n",
       " 'wikipedia.en',\n",
       " 'phosphopoint.kinase',\n",
       " 'phosphopoint.protein',\n",
       " 'inchikey',\n",
       " 'uniprot.isoform',\n",
       " 'kegg.environ',\n",
       " 'cldb',\n",
       " 'efo',\n",
       " 'hgmd',\n",
       " 'aphidbase.transcript',\n",
       " 'affy.probeset',\n",
       " 'treefam',\n",
       " 'caps',\n",
       " 'cubedb',\n",
       " 'ideal',\n",
       " 'stap',\n",
       " 'pocketome',\n",
       " 'gold.genome',\n",
       " 'gold.meta',\n",
       " 'bugbase.protocol',\n",
       " 'bugbase.expt',\n",
       " 'tol',\n",
       " 'vario',\n",
       " 'vipr',\n",
       " 'epd',\n",
       " 'rfam',\n",
       " 'fbol',\n",
       " 'aftol.taxonomy',\n",
       " 'aspgd.locus',\n",
       " 'aspgd.protein',\n",
       " 'autdb',\n",
       " 'bacmap.map',\n",
       " 'bgee.family',\n",
       " 'bgee.gene',\n",
       " 'bgee.stage',\n",
       " 'bgee.organ',\n",
       " 'biocarta.pathway',\n",
       " 'panther.pthcmp',\n",
       " 'gmd.profile',\n",
       " 'gmd.gcms',\n",
       " 'gmd.ref',\n",
       " 'gmd.analyte',\n",
       " 'intact.molecule',\n",
       " 'depod',\n",
       " 'cst',\n",
       " 'cst.ab',\n",
       " 'ndc',\n",
       " 'phytozome.locus',\n",
       " 'subtilist',\n",
       " 'dailymed',\n",
       " 'sider.drug',\n",
       " 'sider.effect',\n",
       " 'wikigenes',\n",
       " 'broad',\n",
       " 'coriell',\n",
       " 'corum',\n",
       " 'ecoliwiki',\n",
       " 'jstor',\n",
       " 'ma',\n",
       " 'uberon',\n",
       " 'unimod',\n",
       " 'vbrc',\n",
       " 'viralzone',\n",
       " 'go.ref',\n",
       " 'rgd.qtl',\n",
       " 'rgd.strain',\n",
       " 'door',\n",
       " 'degradome',\n",
       " 'dbd',\n",
       " 'datf',\n",
       " 'iuphar.ligand',\n",
       " 'molbase',\n",
       " 'yrcpdr',\n",
       " 'yid',\n",
       " 'funcbase.fly',\n",
       " 'funcbase.human',\n",
       " 'funcbase.mouse',\n",
       " 'funcbase.yeast',\n",
       " 'ydpm',\n",
       " 'wormbase.rnai',\n",
       " 'funcat',\n",
       " 'pass2',\n",
       " 'iceberg.element',\n",
       " 'iceberg.family',\n",
       " 'vfdb.genus',\n",
       " 'vfdb.gene',\n",
       " 'mesh.2013',\n",
       " 'kegg.module',\n",
       " 'kegg.disease',\n",
       " 'medlineplus',\n",
       " 'ligandbox',\n",
       " 'glycoepitope',\n",
       " 'jcggdb',\n",
       " 'noncodev4.gene',\n",
       " 'noncodev4.rna',\n",
       " 'oryzabase.gene',\n",
       " 'oryzabase.mutant',\n",
       " 'oryzabase.strain',\n",
       " 'oryzabase.stage',\n",
       " 'otl',\n",
       " 'genewiki',\n",
       " 'paxdb.organism',\n",
       " 'paxdb.protein',\n",
       " 'pdb.ligand',\n",
       " 'merops.inhibitor',\n",
       " 'adw',\n",
       " 'glida.gpcr',\n",
       " 'glida.ligand',\n",
       " 'grsdb',\n",
       " 'fsnp',\n",
       " 'hdr',\n",
       " 'norine',\n",
       " 'ordb',\n",
       " 'odor',\n",
       " 'p3db.protein',\n",
       " 'p3db.site',\n",
       " 'topdb',\n",
       " 'cattleqtldb',\n",
       " 'chickenqtldb',\n",
       " 'pigqtldb',\n",
       " 'sheepqtldb',\n",
       " 'gramene.growthstage',\n",
       " 'ncbigi',\n",
       " 'ebimetagenomics.samp',\n",
       " 'ega.study',\n",
       " 'ega.dataset',\n",
       " 'px',\n",
       " 'biomodels.vocabulary',\n",
       " 'pride.project',\n",
       " 'antibodyregistry',\n",
       " 'mamo',\n",
       " 'eo',\n",
       " 'idot',\n",
       " 'hgnc.family',\n",
       " 'yeastintron',\n",
       " 'ardb',\n",
       " 'proteomicsdb.protein',\n",
       " 'proteomicsdb.peptide',\n",
       " 'hpm.protein',\n",
       " 'hpm.peptide',\n",
       " 'drugbankv4.target',\n",
       " 'zinc',\n",
       " 'foodb.compound',\n",
       " 'unii',\n",
       " 'orphanet.ordo',\n",
       " 'psipar',\n",
       " 'clinvar.record',\n",
       " 'ebimetagenomics.proj',\n",
       " 'euclinicaltrials',\n",
       " 'google.patent',\n",
       " 'uspto',\n",
       " 'cpc',\n",
       " 'gwascentral.study',\n",
       " 'exac.variant',\n",
       " 'gwascentral.marker',\n",
       " 'gwascentral.phenotype',\n",
       " 'lincs.cell',\n",
       " 'lincs.protein',\n",
       " 'lincs.smallmolecule',\n",
       " 'exac.transcript',\n",
       " 'exac.gene',\n",
       " 'wikidata',\n",
       " 'swisslipid',\n",
       " 'unipathway.compound',\n",
       " 'seed',\n",
       " 'seed.compound',\n",
       " 'bigg.model',\n",
       " 'bigg.compartment',\n",
       " 'bigg.metabolite',\n",
       " 'bigg.reaction',\n",
       " 'rrid',\n",
       " 'umls',\n",
       " 'mesh',\n",
       " 'emdb',\n",
       " 'mirtarbase',\n",
       " 'meddra',\n",
       " 'dashr',\n",
       " 'dashr.expression',\n",
       " 'splash',\n",
       " 'metanetx.chemical',\n",
       " 'metanetx.reaction',\n",
       " 'metanetx.compartment',\n",
       " 'unipathway.reaction',\n",
       " 'hp',\n",
       " 'sasbdb',\n",
       " 'hgnc.genefamily',\n",
       " 'mdm',\n",
       " 'probonto',\n",
       " 'apid.interactions',\n",
       " 'storedb',\n",
       " 'kegg',\n",
       " 'planttfdb',\n",
       " 'mp',\n",
       " 'worms',\n",
       " 'multicellds.collection',\n",
       " 'multicellds.snapshot',\n",
       " 'multicellds.cell_line',\n",
       " 'ecyano.entity',\n",
       " 'ecyano.model',\n",
       " 'ecyano.rule',\n",
       " 'lincs.data',\n",
       " 'agricola',\n",
       " 'ms',\n",
       " 'envo',\n",
       " 'ark',\n",
       " 'grid',\n",
       " 'medgen',\n",
       " 'clinvar.submission',\n",
       " 'clinvar',\n",
       " 'bao',\n",
       " 'cellosaurus',\n",
       " 'mir',\n",
       " 'pgx',\n",
       " 'cosmic',\n",
       " 'dbgap',\n",
       " 'iao',\n",
       " 'gdc',\n",
       " 'omit',\n",
       " 'insdc.cds',\n",
       " 'insdc.gca',\n",
       " 'glytoucan',\n",
       " 'metacyc.compound',\n",
       " 'metacyc.reaction',\n",
       " 'd1id',\n",
       " 'cath',\n",
       " 'mmp.ref',\n",
       " 'mmp.db',\n",
       " 'mmp.cat',\n",
       " 'biotools',\n",
       " 'aop',\n",
       " 'ngl',\n",
       " 'erv',\n",
       " 'napdi',\n",
       " 'arraymap',\n",
       " 'begdb',\n",
       " 'cameo',\n",
       " 'sisu',\n",
       " 'mzspec',\n",
       " 'crisprdb',\n",
       " 'gnpis',\n",
       " 'licebase',\n",
       " 'sugarbind',\n",
       " 'rnacentral',\n",
       " 'microscope',\n",
       " 'swissregulon',\n",
       " 'validatordb',\n",
       " 'biominder',\n",
       " 'neurovault.collection',\n",
       " 'vmhmetabolite',\n",
       " 'mobidb',\n",
       " 'neurovault.image',\n",
       " 'nmr',\n",
       " 'vmhreaction',\n",
       " 'ascl',\n",
       " 'fb',\n",
       " 'gudmap',\n",
       " 'eu89h']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prefixes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are #629 Compact ID prefixes registered in identifiers.org\n"
     ]
    }
   ],
   "source": [
    "print(\"There are #{} Compact ID prefixes registered in identifiers.org\".format(len(prefixes)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Ensembl', Resource Information 'Ensembl US West mirror'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Ensembl', Resource Information 'Ensembl US East mirror'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Ensembl', Resource Information 'Ensembl Asia mirror'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Enzyme Nomenclature', Resource Information 'KEGG Ligand Database for Enzyme Nomenclature'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UniProt Knowledgebase', Resource Information 'Universal Protein Resource using Persistent URL system'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Taxonomy', Resource Information 'Taxonomy through UniProt PURL'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Taxonomy', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BioModels Database', Resource Information 'Caltech mirror'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BioModels Database', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ICD', Resource Information 'International Statistical Classification of Diseases and Related Health Problems'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'InterPro', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Pathway', Resource Information 'KEGG PATHWAY Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Compound', Resource Information 'KEGG LIGAND Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Reaction', Resource Information 'KEGG Reaction Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PubMed', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PubMed', Resource Information 'PubMed through Linkedlife data'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'OMIM', Resource Information 'OMIM at John Hopkins'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'OMIM', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'OMIM', Resource Information 'OMIM mirror at John Hopkins'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PIRSF', Resource Information 'PIRSF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Reactome', Resource Information 'Reactome, a curated knowledgebase of biological pathways'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DOI', Resource Information 'Digital Object Identifier'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Protein Data Bank', Resource Information 'Proteopedia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Gene Ontology', Resource Information 'GO Browser'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Gene Ontology', Resource Information 'GO through PANTHER'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SGD', Resource Information 'SGD'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SGD', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Drug', Resource Information 'KEGG DRUG Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Glycan', Resource Information 'KEGG GLYCAN Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'WormBase', Resource Information 'WormBase (Master)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'WormBase', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Nucleotide Sequence Database', Resource Information 'INSDC through GenBank'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Nucleotide Sequence Database', Resource Information 'INSDC through DDBJ'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'FlyBase', Resource Information 'The FlyBase Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Wormpep', Resource Information 'Wormpep (Master)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'arXiv', Resource Information 'Cornell University arXiv'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Mouse Genome Database', Resource Information 'Mouse Genome Informatics Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SABIO-RK Reaction', Resource Information 'SABIO-RK reaction'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Transport Classification Database', Resource Information 'TCDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UniParc', Resource Information 'UniParc through UniProt'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MINT', Resource Information 'The Molecular INTeraction database (MINT)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Database of Interacting Proteins', Resource Information 'Database of interacting proteins'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Signaling Gateway', Resource Information 'Signaling Gateway Molecule Pages'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'RESID', Resource Information 'RESID at Georgetown University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Rat Genome Database', Resource Information 'Rat Genome Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TAIR Protein', Resource Information 'The Arabidopsis Information Resource (TAIR) Protein'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TAIR Gene', Resource Information 'The Arabidopsis Information Resource (TAIR) Gene'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TAIR Locus', Resource Information 'The Arabidopsis Information Resource (TAIR) Locus'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HMDB', Resource Information 'The Human Metabolome Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'LIPID MAPS', Resource Information 'Department of Bioengineering and the San Diego Supercomputer Center'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PeptideAtlas', Resource Information 'PeptideAtlas at ISB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Saccharomyces genome database pathways', Resource Information 'Saccharomyces cerevisiae pathways at SGD.'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BioGRID', Resource Information 'BioGRID database of physical and genetic interactions'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MEROPS', Resource Information 'MEROPS database at Sanger Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PANTHER Family', Resource Information 'PANTHER Family at USC (Los Angeles)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PRINTS', Resource Information 'PRINTS through SPRINT'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Ligand Expo', Resource Information 'Ligand Expo at Rutgers'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Aclame', Resource Information 'Aclame database of mobile genetic elements'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ISBN', Resource Information 'ISBNDB - ISBN Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ISBN', Resource Information 'ISBN database at WorldCat'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name '3DMET', Resource Information '3DMET database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MatrixDB', Resource Information 'MatrixDB Association'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NCBI Gene', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Genes', Resource Information 'KEGG GENES Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BRENDA', Resource Information 'Brenda enzyme database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Pathway Commons', Resource Information 'Pathway Commons'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HOVERGEN', Resource Information 'Hovergen'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Melanoma Molecular Map Project Biomaps', Resource Information 'main webpage'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'WikiPathways', Resource Information 'Wikipathways'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'miRBase Sequence', Resource Information 'miRBase Sequence Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ZFIN Gene', Resource Information 'ZFIN Expression at University of Oregon'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HGNC', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Sequence Ontology', Resource Information 'Sequence Ontology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Rhea', Resource Information 'Rhea'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Unipathway', Resource Information 'UniPathway'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ChEMBL compound', Resource Information 'ChEMBL compound RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ChEMBL compound', Resource Information 'ChEMBL compound RDF through EBI RDF Platform'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ChEMBL target', Resource Information 'ChEMBL target RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ChEMBL target', Resource Information 'ChEMBL target RDF through EBI RDF Platform'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SABIO-RK Kinetic Record', Resource Information 'SABIO-RK Reaction Kinetics Database - Kinetic Dataset'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Anatomical Therapeutic Chemical', Resource Information 'Anatomical Therapeutic Chemical Index at WHO'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PharmGKB Pathways', Resource Information 'PharmGKB at Stanford'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PharmGKB Disease', Resource Information 'PharmGKB Disease at Stanford'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PharmGKB Drug', Resource Information 'PharmGKB Drug at Stanford'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TTD Drug', Resource Information 'Therapeutic Target Database Drug'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TTD Target', Resource Information 'Therapeutic Target Database Target'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NeuronDB', Resource Information 'NeuronDB at SenseLab'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NeuroMorpho', Resource Information 'NeuroMorpho'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ChemIDplus', Resource Information 'ChemIDplus at National Library of Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CTD Chemical', Resource Information 'Comparative Toxicogenomics Database (Chemical)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CTD Disease', Resource Information 'Comparative Toxicogenomics Database (Disease)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CTD Gene', Resource Information 'Comparative Toxicogenomics Database (Gene)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BioNumbers', Resource Information 'BioNumbers database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DrugBank', Resource Information 'DrugBank drug information'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DrugBank', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'T3DB', Resource Information 'Toxin and Toxin Target Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Small Molecule Pathway Database', Resource Information 'Small Molecule Pathway Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PhosphoSite Protein', Resource Information 'PhosphoSite'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GeneDB', Resource Information 'GeneDB at Sanger Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GlycomeDB', Resource Information 'GlycomeDB carbohydrate structure database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'LipidBank', Resource Information 'LipidBank the Lipid database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Orthology', Resource Information 'KEGG Orthology Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ProDom', Resource Information 'ProDom Protein Domain Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SMART', Resource Information 'SMART (Simple Modular Architecture Research Tool)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Relation Ontology', Resource Information 'OBO Relation Ontology through OntoBee'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Molecular Modeling Database', Resource Information 'MMDB at NCBI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'IMEx', Resource Information 'IMEx Consortium though Intact'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'iRefWeb', Resource Information 'iRefWeb Protein Interaction Reference Index'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Microbial Protein Interaction Database', Resource Information 'Microbial Protein Interaction Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PhosphoSite Residue', Resource Information 'PhosphoSite Residue'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NeuroLex', Resource Information 'NeuroLex Neuroscience Lexicon'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Ontology for Biomedical Investigations', Resource Information 'OBI through Ontobee'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SABIO-RK EC Record', Resource Information 'SABIO-RK Reaction Kinetics Database - Enzyme Classification (EC)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'JWS Online', Resource Information 'JWS Online Model Repository at Stellenbosch'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'JWS Online', Resource Information 'JWS Online Model Repository at Amsterdam'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'JWS Online', Resource Information 'JWS Online Model Repository at Manchester'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ModelDB', Resource Information 'ModelDB at SenseLab'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SubtiWiki', Resource Information 'SubtiWiki website'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NCI Pathway Interaction Database: Pathway', Resource Information 'NCI Pathway Interaction Database (Pathway)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Database of Quantitative Cellular Signaling: Model', Resource Information 'Database of Quantitative Cellular Signaling (DOQCS) - Model Access'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Database of Quantitative Cellular Signaling: Pathway', Resource Information 'Database of Quantitative Cellular Signaling (DOQCS) - Pathway Access'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ClinicalTrials.gov', Resource Information 'ClinicalTrials.gov at NIH'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ClinicalTrials.gov', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ChemSpider', Resource Information 'ChemSpider at RSC'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NCIt', Resource Information 'NCIt at National Cancer Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Protein Ontology', Resource Information 'Protein Ontology at Georgetown'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'OMIA', Resource Information 'OMIA at University of Sydney'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ChemBank', Resource Information 'ChemBank at Broad Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Candida Genome Database', Resource Information 'Candida Genome Database at Stanford University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'AntWeb', Resource Information 'AntWeb at California Academy of Sciences'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'AmoebaDB', Resource Information 'AmoebaDB at EuPathDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CryptoDB', Resource Information 'CryptoDB at EuPathDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PlasmoDB', Resource Information 'PlasmoDB at EuPathDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GiardiaDB', Resource Information 'GiardiaDB at EuPathDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MicrosporidiaDB', Resource Information 'MicrosporidiaDB at EuPathDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ToxoDB', Resource Information 'ToxoDB at EuPathDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TrichDB', Resource Information 'TrichDB at EuPathDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TriTrypDB', Resource Information 'TriTrypDB at EuPathDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BDGP insertion DB', Resource Information 'BDGP Gene Disruption Project'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BeetleBase', Resource Information 'BeetleBase at Kansas State University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BOLD Taxonomy', Resource Information 'BOLD taxonomy browser'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'dbEST', Resource Information 'dbEST through DNA Data Bank of Japan (DDBJ)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'EcoGene', Resource Information 'EcoGene at University of Miami'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GABI', Resource Information 'GABI at Max Planck'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GreenGenes', Resource Information 'GreenGenes at Lawrence Berkeley'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GRIN Plant Taxonomy', Resource Information 'GRIN at Maryland'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'H-InvDb Locus', Resource Information 'H-Invitational Locus View at BIRC'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'H-InvDb Transcript', Resource Information 'H-Invitational Transcript View at BIRC'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'H-InvDb Protein', Resource Information 'H-Invitational Protein View at BIRC'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HOMD Sequence Metainformation', Resource Information 'HOMD sequence metainformation at The Forsyth Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HOMD Taxonomy', Resource Information 'HOMD taxon information at The Forsyth Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'IRD Segment Sequence', Resource Information 'IRD at BioHealthBase'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ISFinder', Resource Information 'IS Finder at CNRS'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Japan Collection of Microorganisms', Resource Information 'JCM at RIKEN BioResource Center'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Integrated Microbial Genomes Taxon', Resource Information 'Integrated Microbial Genomes Taxon at Lawrence Berkeley National Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Integrated Microbial Genomes Gene', Resource Information 'Integrated Microbial Genomes Taxon at Lawrence Berkeley National Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MaizeGDB Locus', Resource Information 'MaizeGDB at Missouri'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MycoBank', Resource Information 'MycoBank at Fungal Biodiversity Centre'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NITE Biological Research Center Catalogue', Resource Information 'NBRC Catalogue at National Institute of Technology Evaluation'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Pseudomonas Genome Database', Resource Information 'Pseudomonas Genome Database at Simon Fraser University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Gramene protein', Resource Information 'Gramene Protein at Cold Spring Harbor'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Gramene genes', Resource Information 'Gramene Genes at Cold Spring Harbor'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Gramene Taxonomy', Resource Information 'Gramene Taxonomy at Cold Spring Harbor'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Gramene QTL', Resource Information 'Gramene Quantitative Trait Loci at Cold Spring Harbor'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Sol Genomics Network', Resource Information 'Sol Genomics Network at Boyce Thompson Institute for Plant Research'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Xenbase', Resource Information 'Xenbase at University of Calgary'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Protein Model Database', Resource Information 'Protein Model Database at University of Rome'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ArachnoServer', Resource Information 'ArachnoServer at Brisbane'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BioCyc', Resource Information 'BioCyc at SRI International'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CAZy', Resource Information 'CAZy at CNRS'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PaleoDB', Resource Information 'PaleoDB at Macquarie University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Compulyeast', Resource Information 'Compluyeast database at Universidad Complutense Madrid'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DisProt', Resource Information 'DisProt at Indiana University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'EchoBASE', Resource Information 'EchoBASE at the University of York'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'eggNOG', Resource Information 'eggNOG Version 3 at European Molecular Biology Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HCVDB', Resource Information 'HCVDB at CNRS'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Genatlas', Resource Information 'Genatlas at Paris Descartes University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CATH superfamily', Resource Information 'CATH superfamily at UCL'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CATH domain', Resource Information 'CATH domain at UCL'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GeneFarm', Resource Information 'GeneFarm at Unité de Recherche en Génomique Végétale'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GPCRDB', Resource Information 'GPCRDB at Radboud University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HOGENOM', Resource Information 'HOGENOM at CNRS'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GeneTree', Resource Information 'GeneTree at Ensembl'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HSSP', Resource Information 'HSSP at EMBL'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HSSP', Resource Information 'HSSP at CMBI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MycoBrowser tuberculosis', Resource Information 'MycoBrowser tuberculosis at Global Health Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MycoBrowser leprae', Resource Information 'MycoBrowser leprae at Global Health Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MycoBrowser marinum', Resource Information 'MycoBrowser marinum at Global Health Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MycoBrowser smegmatis', Resource Information 'MycoBrowser smegmatis at Global Health Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Orphanet', Resource Information 'Orphanet at Inserm'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Orphanet', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'OrthoDB', Resource Information 'OrthoDB at Swiss Institute of Bioinformatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Peroxibase', Resource Information 'Peroxibase at University of Geneva'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PhylomeDB', Resource Information 'PhylomeDB at Centro de Investigación Príncipe Felipe'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SubstrateDB', Resource Information 'SubstrateDB at The Burnham Institute for Medical Research'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CutDB', Resource Information 'CutDB at The Burnham Institute for Medical Research'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PMP', Resource Information 'PMP at University of Basel'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ProtoNet ProteinCard', Resource Information 'ProtoNet at Hebrew University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ProtoNet Cluster', Resource Information 'ProtoNet at Hebrew University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'REBASE', Resource Information 'REBASE at New England Biolabs'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SWISS-MODEL', Resource Information 'SWISS-MODEL at SIB Swiss Institute of Bioinformatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Human Disease Ontology', Resource Information 'Human Disease Ontology at Northwestern University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'miRBase mature sequence', Resource Information 'miRBase'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'nextProt', Resource Information 'nextProt at Swiss Institute of Bioinformatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CAS', Resource Information 'CAS through Common Chemistry'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Genome', Resource Information 'KEGG Genome Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Metagenome', Resource Information 'KEGG Metagenome Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NARCIS', Resource Information 'NARCIS at The Hague'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Japan Chemical Substance Dictionary', Resource Information 'Japan Chemical Substance Dictionary'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Pathway Ontology', Resource Information 'Pathway Ontology at Rat Genome Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Sequence Read Archive', Resource Information 'DDBJ Sequence Read Archive (DRA)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ScerTF', Resource Information 'ScerTF at Washington University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PharmGKB Gene', Resource Information 'PharmGKB Gene at Stanford'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'miRNEST', Resource Information 'miRNEST at Adam Mickiewicz University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NAPP', Resource Information 'NAPP at CNRS'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NONCODE v3', Resource Information 'NONCODE at Chinese Academy of Sciences'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'VIRsiRNA', Resource Information 'VIRsiRNA at Institute of Microbial Technology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ELM', Resource Information 'ELM at EMBL (Germany)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MimoDB', Resource Information 'BDB at University of Electronic Science and Technology of China'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SitEx', Resource Information 'SitEx at Novosibirsk State University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BYKdb', Resource Information 'BYKdb at CNRS'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Conoserver', Resource Information 'ConoServer at University of Queensland'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TopFind', Resource Information 'TopFind at University of British Columbia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MIPModDB', Resource Information 'MIPModDB at Indian Institute of Technology Kanpur'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Cell Image Library', Resource Information 'Cell Image Library at American Society for Cell Biology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ZFIN Phenotype', Resource Information 'ZFIN Phenotype at University of Oregon'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ZFIN Expression', Resource Information 'ZFIN Expression at University of Oregon'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CABRI', Resource Information 'CABRI Cell Lines catalogue in Genova (SRS)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CABRI', Resource Information 'CABRI Cell Lines catalogue in Brussels (SRS)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CYGD', Resource Information 'CYGD PEDANT Interface at Biomax Informatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HUGE', Resource Information 'HUGE at Kazusa DNA Research Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BindingDB', Resource Information 'BindingDB At Skaggs School of Pharmacy & Pharmaceutical Sciences'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'STRING', Resource Information 'STRING at Heidelberg'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'STRING', Resource Information 'STRING Mirror at Heidelberg'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'STITCH', Resource Information 'STITCH'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Anatomical Therapeutic Chemical Vetinary', Resource Information 'Anatomical Therapeutic Chemical Vet Index at WHO'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Phenol-Explorer', Resource Information 'Phenol-Explorer Database on Phenol content of foods'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SNOMED CT', Resource Information 'SNOMED-CT at The National Pathology Exchange'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MeSH 2012', Resource Information 'MeSH at National Library of Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KnapSack', Resource Information 'KnapSack at Nara Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Canadian Drug Product Database', Resource Information 'Canadian Drug Identification Number at Health Canada'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MassBank', Resource Information 'MassBank in Japan'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MassBank', Resource Information 'MassBank in Europe'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Golm Metabolome Database', Resource Information 'Golm Metabolome Database at Max Planck Institute of Molecular Plant Physiology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HomoloGene', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UM-BBD Compound', Resource Information 'Biocatalysis/Biodegradation Database Mirror (Compound) at ETH Zurich'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ABS', Resource Information 'ABS at IMIM'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'APD', Resource Information 'APD at Nebraska'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ChemDB', Resource Information 'ChemDB at UC Irvine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DPV', Resource Information 'DPV at Rothamsted'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'IUPHAR receptor', Resource Information 'IUPHAR Receptor at University of Edinburgh'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ASAP', Resource Information 'ASAP at University of Wisconsin'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ATCC', Resource Information 'ATCC in Virginia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Dictybase Gene', Resource Information 'Dictybase Gene at Northwestern University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'IMGT LIGM', Resource Information 'IMGT LIGM through DKFZ (SRS)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'IMGT LIGM', Resource Information 'IMGT LIGM at CNRS'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Worfdb', Resource Information 'Worm Orf DB at Harvard'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NEXTDB', Resource Information 'NEXTDB at Shizuoka'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SoyBase', Resource Information 'SoyBase at Iowa State University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HAMAP', Resource Information 'HAPMAP at Swiss Institute of Bioinformatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Rouge', Resource Information 'Rouge at Kazusa DNA Research Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CGSC Strain', Resource Information 'CGSC at Yale university'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DragonDB DNA', Resource Information 'DragonDB DNA at University of British Columbia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DragonDB Protein', Resource Information 'DragonDB Protein at University of British Columbia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DragonDB Locus', Resource Information 'DragonDB Locus at University of British Columbia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DragonDB Allele', Resource Information 'DragonDB Allele at University of British Columbia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ISSN', Resource Information 'ISSN at Library of Congress'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MEROPS Family', Resource Information 'Merops Family at Sanger Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NASC code', Resource Information 'NASC at University of Nottingham'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NIAEST', Resource Information 'NIAEST at National Institute on Aging'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Pazar Transcription Factor', Resource Information 'Pazar Transcription Factor at University of British Columbia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'RNA Modification Database', Resource Information 'RNA Modification Database at University of Utah'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TreeBASE', Resource Information 'TreeBASE (hosted at National Evolutionary Synthesis Center)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Tetrahymena Genome Database', Resource Information 'TGD at Stanford University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TIGRFAMS', Resource Information 'TIGRFAM at JCVI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Animal TFDB Family', Resource Information 'Animal TFDB at Hubei Bioinformatics & Molecular Imaging Key Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'IUPHAR family', Resource Information 'The IUPHAR/BPS Guide to PHARMACOLOGY'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DBG2 Introns', Resource Information 'Bacterial Group II Introns at University of Calgary'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Spectral Database for Organic Compounds', Resource Information 'SDBS at AIST'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Vbase2', Resource Information 'Vbase2 at German Research Centre for Biotechnology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SPIKE Map', Resource Information 'SPIKE database at Tel Aviv University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'METLIN', Resource Information 'METLIN at Scripps Center, Molecular Biology and Center for Mass Spectrometry, La Jolla'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GeneCards', Resource Information 'GeneCards at Weizmann Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MMRRC', Resource Information 'MMRRC  at University of North Carolina'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UM-BBD Reaction', Resource Information 'Biocatalysis/Biodegradation Database Mirror (Reaction) at ETH Zurich'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UM-BBD Enzyme', Resource Information 'Biocatalysis/Biodegradation Database Mirror (Enzyme) at ETH Zurich'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UM-BBD Pathway', Resource Information 'Biocatalysis/Biodegradation Database Mirror (Pathway) at ETH Zurich'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UM-BBD Biotransformation Rule', Resource Information 'Biocatalysis/Biodegradation Database Mirror (Biotransformation Rule) at ETH Zurich'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'mirEX', Resource Information 'mirEX at Adam Mickiewicz University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Dictybase EST', Resource Information 'Dictybase Gene at Northwestern University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Flystock', Resource Information 'Flystock at Indiana University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'OPM', Resource Information 'OPM at University of Michigan'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Allergome', Resource Information 'Allergome at Rome'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PomBase', Resource Information 'PomBase at University of Cambridge'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HPA', Resource Information 'Human Protein Atlas at AlbaNova University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'JAX Mice', Resource Information 'JAX Mice at Jackson Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'uBio NameBank', Resource Information 'uBio NameBank at The Marine Biological Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'YeTFasCo', Resource Information 'YeTFasCo at University of Toronto'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TarBase', Resource Information 'TarBase v7 at University of Thessaly'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CharProt', Resource Information 'CharProt at JCVI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'OMA Protein', Resource Information 'OMA Protein through OMA browser at ETH Zurich'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'OMA Group', Resource Information 'OMA Group through OMA browser at ETH Zurich'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UniGene', Resource Information 'UniGene at NCBI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BitterDB Receptor', Resource Information 'BitterDB Receptor at The Hebrew University of Jerusalem'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BitterDB Compound', Resource Information 'BitterDB Compound at The Hebrew University of Jerusalem'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BioProject', Resource Information 'BioProject at DNA Data Bank of Japan'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BioSample', Resource Information 'BioSample at DNA Data Bank of Japan'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PiroplasmaDB', Resource Information 'PiroplasmaBD at EuPathDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Unite', Resource Information 'Unite at University of Tartu'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NCIm', Resource Information 'NCIm at National Cancer Institue'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ProGlycProt', Resource Information 'ProGlycProt at IMT'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PolBase', Resource Information 'PolBase at New England BioLabs'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NucleaRDB', Resource Information 'NucleaRDB at Radboud University Nijmegen Medical Centre'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SUPFAM', Resource Information 'SUPFAM at MRC and University of Bristol'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Rice Genome Annotation Project', Resource Information 'Rice Genome Annotation Project at TIGR'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PINA', Resource Information 'Protein Interaction Network Analysis (PINA) at Garvan Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Tissue List', Resource Information 'Tissue List at Swiss Institute of Bioinformatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BacMap Biography', Resource Information 'BacMap Biography at University of Alberta'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PANTHER Pathway', Resource Information 'PANTHER Pathway at USC (Los Angeles)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'FAIRsharing', Resource Information 'FAIRSharing at University of Oxford'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'FungiDB', Resource Information 'FungiDB at University of California'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DARC', Resource Information 'DARC at University of Munich'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DRSC', Resource Information 'DRSC at Harvard Medical School'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'OriDB Schizosaccharomyces', Resource Information 'OriDB Schizosaccharomyces at University of Nottingham'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'OriDB Saccharomyces', Resource Information 'OriDB Saccharomyces at University of Nottingham'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PSCDB', Resource Information 'PSCDB at Nagoya University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SCOP', Resource Information 'SCOP at MRC'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SCOP', Resource Information 'SCOP at Berkeley'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DOMMINO', Resource Information 'DOMMINO at University of Missouri'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PANTHER Node', Resource Information 'PANTHER Node at USC (Los Angeles)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Locus Reference Genomic', Resource Information 'Locus Reference Genomic at University of Leicester'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Locus Reference Genomic', Resource Information 'Locus Reference Genomic through Ensembl'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Locus Reference Genomic', Resource Information 'Locus Reference Genomic through Ensembl mirror (asia)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Locus Reference Genomic', Resource Information 'Locus Reference Genomic through Ensembl mirror (US west)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Locus Reference Genomic', Resource Information 'Locus Reference Genomic through Ensembl mirror (US east)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HPRD', Resource Information 'Human Protein Reference Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'National Bibliography Number', Resource Information 'NBN through URN Resolver at German National Library'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ORCID', Resource Information 'ORCID at Bethesda'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'InChI', Resource Information 'InChI through RDF Open Molecules'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'InChI', Resource Information 'InChI through Chemspider'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'InChI', Resource Information 'InChI through NIST'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Wikipedia (En)', Resource Information 'Wikipedia (English)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Wikipedia (En)', Resource Information 'Wikipedia structured content through DBpedia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PhosphoPoint Kinase', Resource Information 'PhosphoPoint Kinase at National Taiwan University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PhosphoPoint Phosphoprotein', Resource Information 'PhosphoPoint Phosphoprotein at National Taiwan University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'InChIKey', Resource Information 'InChIKey through ChemSpider'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'InChIKey', Resource Information 'InChiKey resolver at NCI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UniProt Isoform', Resource Information 'UniProt Isoform through Universal Protein Resource'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UniProt Isoform', Resource Information 'UniProt Isoform through Universal Protein Resource using Persistent URL system'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UniProt Isoform', Resource Information 'UniProt Isoform through UniParc'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Environ', Resource Information 'KEGG ENVIRON Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CLDB', Resource Information 'CLDB at Genova'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HGMD', Resource Information 'HGMD at Cardiff University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'AphidBase Transcript', Resource Information 'AphidBase at INRA'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Affymetrix Probeset', Resource Information 'Affymetrix ProbeSet in Santa Clara'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Affymetrix Probeset', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TreeFam', Resource Information 'TreeFam database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CAPS-DB', Resource Information 'CAPS-DB at Leeds Institute of Molecular Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Cube db', Resource Information 'Cube db at Bioinformatics Institute (Singapore)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'IDEAL', Resource Information 'IDEAL at Maebashi Institute of Technology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'STAP', Resource Information 'STAP at Korean Bioinformation Center'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Pocketome', Resource Information 'Pocketome at UCSD Skaggs School'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GOLD genome', Resource Information 'GOLD genome at Department of Energy Joint Genome Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GOLD metadata', Resource Information 'GOLD metadata at Department of Energy Joint Genome Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BugBase Protocol', Resource Information 'BugBase Protocol at University of London'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BugBase Expt', Resource Information 'BugBase Expt at University of London'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Tree of Life', Resource Information 'Tree of Life Web'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'VariO', Resource Information 'VariO at Lund University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ViPR Strain', Resource Information 'ViPR Strain at J. Craig Venter Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'EPD', Resource Information 'EPD at Swiss Institute of Bioinformatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Fungal Barcode', Resource Information 'Fungal Barcode at Fungal Biodiversity Centre'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'AFTOL', Resource Information 'AFTOL at University of Minnesota'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'AspGD Locus', Resource Information 'AspGD at Stanford Medical School'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'AspGD Protein', Resource Information 'AspGD Protein at Stanford Medical School'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'AutDB', Resource Information 'AutDB at MindSpec'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BacMap Map', Resource Information 'BacMap Genome Map at University of Alberta'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Bgee family', Resource Information 'Bgee at Lausanne'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Bgee gene', Resource Information 'Bgee at Lausanne'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Bgee stage', Resource Information 'Bgee at Lausanne'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Bgee organ', Resource Information 'Bgee at Lausanne'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BioCarta Pathway', Resource Information 'BioCarta Pathway at NCI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PANTHER Pathway Component', Resource Information 'PANTHER Pathway Component at USC (Los Angeles)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Golm Metabolome Database Profile', Resource Information 'Golm Metabolome Database (Profile) at Max Planck Institute of Molecular Plant Physiology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Golm Metabolome Database GC-MS spectra', Resource Information 'Golm Metabolome Database (GC-MS Spectra) at Max Planck Institute of Molecular Plant Physiology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Golm Metabolome Database Reference Substance', Resource Information 'Golm Metabolome Database (Reference Substance) at Max Planck Institute of Molecular Plant Physiology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Golm Metabolome Database Analyte', Resource Information 'Golm Metabolome Database (Analyte) at Max Planck Institute of Molecular Plant Physiology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DEPOD', Resource Information 'DEPOD at EMBL'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Cell Signaling Technology Pathways', Resource Information 'CST Pathways at Cell Signaling Technology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Cell Signaling Technology Antibody', Resource Information 'CST Antibody at Cell Signaling Technology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'National Drug Code', Resource Information 'National Drug Code at Food and Drug Administration'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'National Drug Code', Resource Information 'Bio2RDF'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Phytozome Locus', Resource Information 'Phytozome Locus at Joint Genome Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SubtiList', Resource Information 'SubtiList at Pasteur Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DailyMed', Resource Information 'DailyMed at NLM'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SIDER Drug', Resource Information 'SIDER Drug v2 at EMBL (Heidelberg)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SIDER Side Effect', Resource Information 'SIDER Side Effect v2 at EMBL (Heidelberg)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'WikiGenes', Resource Information 'WikiGenes at Massachusetts Institute of Technology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Broad Fungal Genome Initiative', Resource Information 'Broad Fungal Genome Initiative at Broad Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Coriell Cell Repositories', Resource Information 'Coriell Cell Repositories at Coriell Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CORUM', Resource Information 'CORUM at Institute for Bioinformatics and Systems Biology (Germany)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'EcoliWiki', Resource Information 'EcoliWiki at Texas Agrilife Research'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'JSTOR', Resource Information 'JSTOR Online'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Mouse Adult Gross Anatomy', Resource Information 'Mouse Adult Gross Anatomy at The Jackson Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Unimod', Resource Information 'Unimod at Matrix Sciences Ltd.'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'VBRC', Resource Information 'VBRC at University of Alabama'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ViralZone', Resource Information 'ViralZone at SIB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Gene Ontology Reference', Resource Information 'Gene Ontology Reference at The Gene Ontology Consortium'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Rat Genome Database qTL', Resource Information 'Rat Genome Database qTL at Medical College of Wisconsin'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Rat Genome Database strain', Resource Information 'Rat Genome Database strain at Medical College of Wisconsin'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DOOR', Resource Information 'DOOR v1 at University of Georgia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Degradome Database', Resource Information 'Degradome Database at'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DBD', Resource Information 'DBD at MRC Laboratory of Molecular Biology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DATF', Resource Information 'DATF through PlantTFDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'IUPHAR ligand', Resource Information 'IUPHAR ligand at University of Edinburgh'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Molbase', Resource Information 'Molbase at Chinese Academy of Sciences'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'YRC PDR', Resource Information 'YRC PDR at University of Washington'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Yeast Intron Database v3', Resource Information 'Yeast Intron Database version 3 at Baskin School of Engineering'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'FuncBase Fly', Resource Information 'FuncBase Fly at Harvard Medical School'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'FuncBase Human', Resource Information 'FuncBase Human at Harvard Medical School'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'FuncBase Mouse', Resource Information 'FuncBase Mouse at Harvard Medical School'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'FuncBase Yeast', Resource Information 'FuncBase Yeast at Harvard Medical School'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'YDPM', Resource Information 'YDPM at Stanford University School of Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'WormBase RNAi', Resource Information 'WormBase RNAi at Cold Spring Harbor Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'FunCat', Resource Information 'FunCat at MIPS (Neuherberg)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PASS2', Resource Information 'PASS2 at National centre for Biological Sciences (India)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ICEberg element', Resource Information 'ICEberg element at Shanghai Jiaotong University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ICEberg family', Resource Information 'ICEberg family at Shanghai Jiaotong University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'VFDB Genus', Resource Information 'VFDB Genus at Institute of Pathogen Biology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'VFDB Gene', Resource Information 'VFDB Gene at Institute of Pathogen Biology'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MeSH 2013', Resource Information 'MeSH 2013 at National Library of Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Module', Resource Information 'KEGG Module at Kyoto University Bioinformatics Center'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'KEGG Disease', Resource Information 'KEGG Disease at Kyoto University Bioinformatics Center'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MedlinePlus', Resource Information 'MedlinePlus at NCBI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'LigandBox', Resource Information 'LigandBox at'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GlycoEpitope', Resource Information 'GlycoEpitope at Ritsumeikan University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'JCGGDB', Resource Information 'JCGGDB at Advanced Science Institute (Japan)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NONCODE v4 Gene', Resource Information 'NONCODE v4 Gene at Chinese Academy of Sciences'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NONCODE v4 Transcript', Resource Information 'NONCODE v4 Transcript at Chinese Academy of Sciences'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Oryzabase Gene', Resource Information 'Oryzabase v4 Gene at National Institute of Genetics (Japan)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Oryzabase Mutant', Resource Information 'Oryzabase v4 Mutant at National Institute of Genetics (Japan)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Oryzabase Strain', Resource Information 'Oryzabase v4 Strain at National Institute of Genetics (Japan)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Oryzabase Stage', Resource Information 'Oryzabase v4 Stage at National Institute of Genetics (Japan)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Oryza Tag Line', Resource Information 'Oryza Tag Line at CIRAD-INRA-SUPAGRO-UMI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Gene Wiki', Resource Information 'Gene Wiki through BioGPS'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PaxDb Organism', Resource Information 'PaxDb v3 at University of Zurich'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'PaxDb Protein', Resource Information 'PaxDb v3 at University of Zurich'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Protein Data Bank Ligand', Resource Information 'PDB Ligand at Protein Databank in Europe (PDBe)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MEROPS Inhibitor', Resource Information 'MEROPS Inhibitor at Sanger Institute'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Animal Diversity Web', Resource Information 'Animal Diversity Web at University of Michigan'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GLIDA GPCR', Resource Information 'GLIDA GPCR at Kyoto University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GLIDA Ligand', Resource Information 'GLIDA Ligand at Kyoto University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GRSDB', Resource Information 'GRSDB v2 at  Ramapo College of New Jersey'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'F-SNP', Resource Information 'F-SNP at Queen's University (Canada)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Homeodomain Research', Resource Information 'Homeodomain Research at National Human Genome Research Institute (NIH)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NORINE', Resource Information 'NORINE at Computer Science Laboratory of Lille'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Olfactory Receptor Database', Resource Information 'Olfactory Receptor Database at Yale University School of Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Odor Molecules DataBase', Resource Information 'Odor Molecules DataBase at Yale University School of Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'P3DB Protein', Resource Information 'P3DB Protein at University of Missouri'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'P3DB Site', Resource Information 'P3DB Site at University of Missouri'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'TOPDB', Resource Information 'TOPDB at Hungarian Academy of Sciences'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Animal Genome Cattle QTL', Resource Information 'Animal QTL Cattle at Iowa State University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Animal Genome Chicken QTL', Resource Information 'Animal QTL Chicken at Iowa State University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Animal Genome Pig QTL', Resource Information 'Animal QTL Pig at Iowa State University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Animal Genome Sheep QTL', Resource Information 'Animal QTL Sheep at Iowa State University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Gramene Growth Stage Ontology', Resource Information 'Gramene Growth Stage Ontology at Cold Spring Harbor'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ProteomeXchange', Resource Information 'ProteomeXchange at Seattle'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SBML RDF Vocabulary', Resource Information 'Vocabulary via BioModels.net'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Antibody Registry', Resource Information 'Antibody Registry at University of California'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Plant Environment Ontology', Resource Information 'Plant Environment Ontology through Gramene'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Identifiers.org Terms', Resource Information 'IdoT via biomodels.net'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HGNC Family', Resource Information 'HGNC Family at HUGO Genome Nomenclature Committee'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Yeast Intron Database v4.3', Resource Information 'Yeast Intron Database version 4.3 at Baskin School of Engineering'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Antibiotic Resistance Genes Database', Resource Information 'ARDB at University of Maryland'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ProteomicsDB Protein', Resource Information 'Proteomics DB Protein at Center for Integrated Protein Science'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ProteomicsDB Peptide', Resource Information 'Proteomics DB Peptide at Center for Integrated Protein Science'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Human Proteome Map Protein', Resource Information 'Human Proteome Map Protein at Institute of Bioinformatics (Bangalore)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Human Proteome Map Peptide', Resource Information 'Human Proteome Map Peptide at Institute of Bioinformatics (Bangalore)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DrugBank Target v4', Resource Information 'DrugBank Target information version 4'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ZINC', Resource Information 'ZINC at University of California (San Francisco)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'FooDB Compound', Resource Information 'FooDB database of food additives at University of Alberta'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UNII', Resource Information 'UNII at U.S. Food and Drug Administration'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'EU Clinical Trials', Resource Information 'EU Clinical Trials at  European Medicines Agency'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Google Patents', Resource Information 'Google Patents at Google'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'USPTO', Resource Information 'USPTO patents at USPTO'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Cooperative Patent Classification', Resource Information 'Cooperative Patent Classification at Espace'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GWAS Central Study', Resource Information 'GWAS Central Study at University of Leicester'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ExAC Variant', Resource Information 'ExAC Variant at Exome Aggregation Consortium'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GWAS Central Marker', Resource Information 'GWAS Central Marker at University of Leicester'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GWAS Central Phenotype', Resource Information 'GWAS Central Phenotype at University of Leicester'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'LINCS Cell', Resource Information 'LINCS Portal'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'LINCS Protein', Resource Information 'LINCS Protein at Harvard Medical School'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'LINCS Small Molecule', Resource Information 'LINCS Portal'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ExAC Transcript', Resource Information 'ExAC Transcript at Exome Aggregation Consortium'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ExAC Gene', Resource Information 'ExAC Gene at Exome Aggregation Consortium'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Wikidata', Resource Information 'Wikidata at Wikimedia Foundation'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SwissLipids', Resource Information 'SwissLipids at Swiss Institute of Bioinformatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UniPathway Compound', Resource Information 'UniPathway Compound at Swiss Institute of Bioinformatics (SIB)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SEED Subsystem', Resource Information 'SEED Subsystem at Argonne National Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SEED Compound', Resource Information 'SEED Compound at Argonne National Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BiGG Model', Resource Information 'BiGG Model at University of California'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BiGG Compartment', Resource Information 'BiGG Compartment at University of California'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BiGG Metabolite', Resource Information 'BiGG Metabolite at University of Cakifornia'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BiGG Reaction', Resource Information 'BiGG Reaction at University of California'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'RRID', Resource Information 'RRID at SciCrunch'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UMLS', Resource Information 'Unified Medical Language System at LinkedLifeData'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MeSH', Resource Information 'MeSH Linked Data at National Library of Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'miRTarBase', Resource Information 'miRTarBase at National Chiao Tung University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MedDRA', Resource Information 'MedDRA through BioPortal'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DASHR', Resource Information 'DASHR at  University of Pennsylvania'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DASHR expression', Resource Information 'DASHR expression at University of Pennsylvania'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SPLASH', Resource Information 'Mass Bank of North America (MoNA)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MetaNetX chemical', Resource Information 'MetaNetX chemical at SIB Swiss Institute of Bioinformatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MetaNetX reaction', Resource Information 'MetaNetX reaction at SIB Swiss Institute of Bioinformatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MetaNetX compartment', Resource Information 'MetaNetX compartment at SIB Swiss Institute of Bioinformatics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'UniPathway Reaction', Resource Information 'UniPathway Reaction at Swiss Institute of Bioinformatics (SIB)'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Human Phenotype Ontology', Resource Information 'Human Phenotype Ontology at Institute for Medical Genetics and Human Genetics'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SASBDB', Resource Information 'SASBDB at European Molecular Biology Laboratory'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'HGNC gene family', Resource Information 'HGNC gene family at HUGO Genome Nomenclature Committee'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MDM', Resource Information 'MDM at University of Muenster'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'APID Interactomes', Resource Information 'APID at Salamanca'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'STOREDB', Resource Information 'STOREDB at University of Cambridge'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Kyoto Encyclopedia of Genes and Genomes', Resource Information 'KEGG at Kyoto University Bioinformatics Center'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Plant Transcription Factor Database', Resource Information 'PlantTFDB at Peking University'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Mammalian Phenotype Ontology', Resource Information 'Mammalian Phenotype Ontology at The Jackson Lab'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'World Register of Marine Species', Resource Information 'World Register of Marine Species'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MultiCellDS collection', Resource Information 'MultiCellDScollection at Keck School of Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MultiCellDS Digital snapshot', Resource Information 'MultiCellDS snapshot at Keck School of Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MultiCellDS Digital Cell Line', Resource Information ' MultiCellDS Digital Cell Line at Keck School of Medicine'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'E-cyanobacterium entity', Resource Information 'e-cyanobacterium entity at Masaryk University Brno'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'E-cyanobacterium model', Resource Information 'e-cyanobacterium entity at Masaryk University Brno'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'E-cyanobacterium rule', Resource Information 'e-cyanobacterium entity at Masaryk University Brno'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'LINCS Data', Resource Information 'LINCS Data at University of Miami'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'AGRICOLA', Resource Information 'AGRICOLA at National Agricultural Library'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ARK', Resource Information 'ARK via the Name-to-Thing resolver.'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GRID', Resource Information 'GRID at Digital Science & Research Ltd.'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Progenetix', Resource Information 'Progenetix'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'COSMIC Gene', Resource Information 'COSMIC Gene at Sanger'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Information Artifact Ontology', Resource Information 'IAO through Ontobee'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Genomic Data Commons Data Portal', Resource Information 'Genomic Data Commons'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'INSDC CDS', Resource Information 'INSDC CDS through ENA'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'INSDC CDS', Resource Information 'INSDC CDS through NCBI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'INSDC CDS', Resource Information 'INSDC CDS through DDBJ'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Genome assembly database', Resource Information 'Genome assembly database through ENA'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Genome assembly database', Resource Information 'Genome assembly database NCBI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GlyTouCan', Resource Information 'The Glycan Repository'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MetaCyc Compound', Resource Information 'MetaCyc Metabolic Pathway Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MetaCyc Reaction', Resource Information 'MetaCyc Metabolic Pathway Database - Reaction'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'DataONE', Resource Information 'DataONE'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CATH Protein Structural Domain Superfamily', Resource Information 'CATH Node through UCL'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MarRef', Resource Information 'MarRef through SfB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MarDB', Resource Information 'MarDB through SfB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MarCat', Resource Information 'MarCat through SfB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'BioTools', Resource Information 'BioTools'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'AOPWiki', Resource Information 'AOPWiki'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NASA GeneLab', Resource Information 'NASA GeneLab'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Human Endogenous Retrovirus Database', Resource Information 'Human Endogenous Retrovirus Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Natural Product-Drug Interaction Research Data Repository', Resource Information 'Natural Product-Drug Interaction Research Data Repository'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ArrayMap', Resource Information 'ArrayMap'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Benchmark Energy & Geometry Database', Resource Information 'Benchmark Energy & Geometry Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CAMEO', Resource Information 'CAMEO'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SISu', Resource Information 'SISu'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Universal Spectrum Identifier', Resource Information 'Universal Spectrum Identifier through Peptide Atlas'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'CRISPRdb', Resource Information 'CRISPRdb'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GnpIS', Resource Information 'GnpIS'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'LiceBase', Resource Information 'LiceBase'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SugarBind', Resource Information 'SugarBind'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'RNAcentral', Resource Information 'RNAcentral through EMBL-EBI'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MicroScope', Resource Information 'MicroScope'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'SwissRegulon', Resource Information 'SwissRegulon'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'ValidatorDB', Resource Information 'ValidatorDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Bio-MINDER Tissue Database', Resource Information 'Bio-MINDER Tissue Database'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NeuroVault Collection', Resource Information 'NeuroVault Collection'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'VMH metabolite', Resource Information 'VMH metabolite'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'MobiDB', Resource Information 'MobiDB'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'NeuroVault Image', Resource Information 'NeuroVault'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'VMH reaction', Resource Information 'VMH reaction'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'Astrophysics Source Code Library', Resource Information 'Astrophysics Source Code Library'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'FaceBase Data Repository', Resource Information 'FaceBase Data Repository'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'GUDMAP', Resource Information 'GUDMAP'\n",
      "NO RESOURCE PREFIX FOR: PID Entry Name 'JRC Data Catalogue', Resource Information 'JRC Data Catalogue'\n"
     ]
    }
   ],
   "source": [
    "# Check the distribution of resources\n",
    "resource_prefixes = []\n",
    "for pid_entry in resolver_dump:\n",
    "    if 'resources' in pid_entry:\n",
    "        for resource in pid_entry['resources']:\n",
    "            if 'resourcePrefix' in resource:\n",
    "                resource_prefixes.append(resource['resourcePrefix'])\n",
    "            else:\n",
    "                print(\"NO RESOURCE PREFIX FOR: PID Entry Name '{}', Resource Information '{}'\".format(pid_entry['name'], resource['info']))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ebi',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ebi',\n",
       " 'intenz',\n",
       " 'expasy',\n",
       " 'expenz',\n",
       " 'enzymeportal',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'hubmed',\n",
       " 'epmc',\n",
       " 'rcsb',\n",
       " 'pdbe',\n",
       " 'pdbj',\n",
       " 'ebi',\n",
       " 'quickgo',\n",
       " 'amigo',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'ebi',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'expasy',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'rcsb',\n",
       " 'ebi',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'ols',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'ols',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'bptl',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ebi',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'epmc',\n",
       " 'cpmc',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'bptl',\n",
       " 'ebi',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'quickgo',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'ols',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'bptl',\n",
       " 'ebi',\n",
       " 'amigo',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ebi',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'rcsb',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'ols',\n",
       " 'ols',\n",
       " 'ncbi',\n",
       " 'ebi',\n",
       " 'ebi',\n",
       " 'ols',\n",
       " 'ols',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'bptl',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'ncbi',\n",
       " 'bptl',\n",
       " 'ols',\n",
       " 'sib',\n",
       " 'ebi',\n",
       " 'ncbi',\n",
       " 'ols',\n",
       " 'ols']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resource_prefixes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "resource_prefixes_distribution = Counter(resource_prefixes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Counter({'amigo': 2,\n",
       "         'bptl': 34,\n",
       "         'cpmc': 1,\n",
       "         'ebi': 56,\n",
       "         'enzymeportal': 1,\n",
       "         'epmc': 2,\n",
       "         'expasy': 2,\n",
       "         'expenz': 1,\n",
       "         'hubmed': 1,\n",
       "         'intenz': 1,\n",
       "         'ncbi': 33,\n",
       "         'ols': 39,\n",
       "         'pdbe': 1,\n",
       "         'pdbj': 1,\n",
       "         'quickgo': 2,\n",
       "         'rcsb': 3,\n",
       "         'sib': 1})"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "resource_prefixes_distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There #17 Resource Selectors in identifiers.org\n"
     ]
    }
   ],
   "source": [
    "print(\"There #{} Resource Selectors in identifiers.org\".format(len(resource_prefixes_distribution.keys())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "labels = sorted(resource_prefixes_distribution.keys())\n",
    "values = [resource_prefixes_distribution.get(key) for key in labels]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "unsupported operand type(s) for -: 'str' and 'float'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-17-dedb96371e00>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mfig1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0max1\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mplt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msubplots\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mfig1\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset_size_inches\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m18\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m5\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0max1\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlabels\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0;36m1.5\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolor\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"blue\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m \u001b[0;31m#ax1.pie(values, labels=labels, shadow=True, startangle=90)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;31m#ax1.axis('equal')\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/sw/lib/python3.6/site-packages/matplotlib/__init__.py\u001b[0m in \u001b[0;36minner\u001b[0;34m(ax, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1890\u001b[0m                     warnings.warn(msg % (label_namer, func.__name__),\n\u001b[1;32m   1891\u001b[0m                                   RuntimeWarning, stacklevel=2)\n\u001b[0;32m-> 1892\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0max\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1893\u001b[0m         \u001b[0mpre_doc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0minner\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__doc__\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1894\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mpre_doc\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/sw/lib/python3.6/site-packages/matplotlib/axes/_axes.py\u001b[0m in \u001b[0;36mbar\u001b[0;34m(self, left, height, width, bottom, **kwargs)\u001b[0m\n\u001b[1;32m   2103\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0malign\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'center'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2104\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0morientation\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'vertical'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2105\u001b[0;31m                 \u001b[0mleft\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mleft\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mwidth\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m/\u001b[0m \u001b[0;36m2.\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mxrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mleft\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2106\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0morientation\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'horizontal'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2107\u001b[0m                 bottom = [bottom[i] - height[i] / 2.\n",
      "\u001b[0;32m/sw/lib/python3.6/site-packages/matplotlib/axes/_axes.py\u001b[0m in \u001b[0;36m<listcomp>\u001b[0;34m(.0)\u001b[0m\n\u001b[1;32m   2103\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0malign\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'center'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2104\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0morientation\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'vertical'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2105\u001b[0;31m                 \u001b[0mleft\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mleft\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mwidth\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m/\u001b[0m \u001b[0;36m2.\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mxrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mleft\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2106\u001b[0m             \u001b[0;32melif\u001b[0m \u001b[0morientation\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'horizontal'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2107\u001b[0m                 bottom = [bottom[i] - height[i] / 2.\n",
      "\u001b[0;31mTypeError\u001b[0m: unsupported operand type(s) for -: 'str' and 'float'"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAABBkAAAEzCAYAAAB0cNsFAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAEnJJREFUeJzt3V+I5Xd5x/HPY7axYNWCu4WS3ZhA1+rWFrRDsHhRQVs2\nudi96B8SCP4huDdNaasUIi0q8UqlFgrpny2KVqhp6kUZ6JZctCmCGMmIbTCRlCFtzaaFrDbmRjRN\n+/RijmVcd3eOm2dmzrivFyyc3+98zznPzZfZfe/v/Ka6OwAAAAAv1kv2ewAAAADgh4PIAAAAAIwQ\nGQAAAIARIgMAAAAwQmQAAAAARogMAAAAwIgdI0NVfaKqnqmqr1zm+aqqP6qqzap6tKreOD8mAAAA\nsOqWuZLhk0lOXuH5W5McX/w5k+RPXvxYAAAAwEGzY2To7s8l+a8rLDmd5C96y8NJfryqfnJqQAAA\nAOBgmLgnww1Jntp2fH5xDgAAALiGHNrLD6uqM9n6SkVe9rKX/fxrX/vavfx4AAAAYAdf+tKXvt7d\nR67mtROR4ekkx7YdH12c+z7dfTbJ2SRZW1vrjY2NgY8HAAAAplTVv1/taye+LrGe5O2L3zLxpiTP\ndfd/DrwvAAAAcIDseCVDVX0myVuSHK6q80k+kORHkqS7/zTJuSS3JdlM8q0k79qtYQEAAIDVtWNk\n6O47dni+k/zG2EQAAADAgTTxdQkAAAAAkQEAAACYITIAAAAAI0QGAAAAYITIAAAAAIwQGQAAAIAR\nIgMAAAAwQmQAAAAARogMAAAAwAiRAQAAABghMgAAAAAjRAYAAABghMgAAAAAjBAZAAAAgBEiAwAA\nADBCZAAAAABGiAwAAADACJEBAAAAGCEyAAAAACNEBgAAAGCEyAAAAACMEBkAAACAESIDAAAAMEJk\nAAAAAEaIDAAAAMAIkQEAAAAYITIAAAAAI0QGAAAAYITIAAAAAIwQGQAAAIARIgMAAAAwQmQAAAAA\nRogMAAAAwAiRAQAAABghMgAAAAAjRAYAAABghMgAAAAAjBAZAAAAgBEiAwAAADBCZAAAAABGiAwA\nAADACJEBAAAAGCEyAAAAACNEBgAAAGCEyAAAAACMEBkAAACAEUtFhqo6WVVPVNVmVd1ziedvrKqH\nqurLVfVoVd02PyoAAACwynaMDFV1XZL7ktya5ESSO6rqxEXLfj/JA939hiS3J/nj6UEBAACA1bbM\nlQy3JNns7ie7+/kk9yc5fdGaTvKKxeNXJvmPuREBAACAg2CZyHBDkqe2HZ9fnNvug0nurKrzSc4l\n+c1LvVFVnamqjarauHDhwlWMCwAAAKyqqRs/3pHkk919NMltST5dVd/33t19trvXunvtyJEjQx8N\nAAAArIJlIsPTSY5tOz66OLfdXUkeSJLu/kKSH01yeGJAAAAA4GBYJjI8kuR4Vd1cVddn68aO6xet\n+VqStyZJVb0uW5HB9yEAAADgGrJjZOjuF5LcneTBJF/N1m+ReKyq7q2qU4tl703y7qr65ySfSfLO\n7u7dGhoAAABYPYeWWdTd57J1Q8ft596/7fHjSd48OxoAAABwkEzd+BEAAAC4xokMAAAAwAiRAQAA\nABghMgAAAAAjRAYAAABghMgAAAAAjBAZAAAAgBEiAwAAADBCZAAAAABGiAwAAADACJEBAAAAGCEy\nAAAAACNEBgAAAGCEyAAAAACMEBkAAACAESIDAAAAMEJkAAAAAEaIDAAAAMAIkQEAAAAYITIAAAAA\nI0QGAAAAYITIAAAAAIwQGQAAAIARIgMAAAAwQmQAAAAARogMAAAAwAiRAQAAABghMgAAAAAjRAYA\nAABghMgAAAAAjBAZAAAAgBEiAwAAADBCZAAAAABGiAwAAADACJEBAAAAGCEyAAAAACNEBgAAAGCE\nyAAAAACMEBkAAACAESIDAAAAMEJkAAAAAEaIDAAAAMAIkQEAAAAYITIAAAAAI0QGAAAAYMRSkaGq\nTlbVE1W1WVX3XGbNr1fV41X1WFX95eyYAAAAwKo7tNOCqrouyX1JfinJ+SSPVNV6dz++bc3xJO9L\n8ubufraqfmK3BgYAAABW0zJXMtySZLO7n+zu55Pcn+T0RWveneS+7n42Sbr7mdkxAQAAgFW3TGS4\nIclT247PL85t95okr6mqz1fVw1V1cmpAAAAA4GDY8esSP8D7HE/yliRHk3yuqn62u7+5fVFVnUly\nJkluvPHGoY8GAAAAVsEyVzI8neTYtuOji3PbnU+y3t3/3d3/muRfshUdvkd3n+3ute5eO3LkyNXO\nDAAAAKygZSLDI0mOV9XNVXV9ktuTrF+05m+ydRVDqupwtr4+8eTgnAAAAMCK2zEydPcLSe5O8mCS\nryZ5oLsfq6p7q+rUYtmDSb5RVY8neSjJ73b3N3ZraAAAAGD1VHfvywevra31xsbGvnw2AAAAcGlV\n9aXuXrua1y7zdQkAAACAHYkMAAAAwAiRAQAAABghMgAAAAAjRAYAAABghMgAAAAAjBAZAAAAgBEi\nAwAAADBCZAAAAABGiAwAAADACJEBAAAAGCEyAAAAACNEBgAAAGCEyAAAAACMEBkAAACAESIDAAAA\nMEJkAAAAAEaIDAAAAMAIkQEAAAAYITIAAAAAI0QGAAAAYITIAAAAAIwQGQAAAIARIgMAAAAwQmQA\nAAAARogMAAAAwAiRAQAAABghMgAAAAAjRAYAAABghMgAAAAAjBAZAAAAgBEiAwAAADBCZAAAAABG\niAwAAADACJEBAAAAGCEyAAAAACNEBgAAAGCEyAAAAACMEBkAAACAESIDAAAAMEJkAAAAAEaIDAAA\nAMAIkQEAAAAYITIAAAAAI0QGAAAAYMRSkaGqTlbVE1W1WVX3XGHdr1RVV9Xa3IgAAADAQbBjZKiq\n65Lcl+TWJCeS3FFVJy6x7uVJfivJF6eHBAAAAFbfMlcy3JJks7uf7O7nk9yf5PQl1n0oyYeTfHtw\nPgAAAOCAWCYy3JDkqW3H5xfn/l9VvTHJse7+28HZAAAAgAPkRd/4sapekuRjSd67xNozVbVRVRsX\nLlx4sR8NAAAArJBlIsPTSY5tOz66OPddL0/y+iT/WFX/luRNSdYvdfPH7j7b3WvdvXbkyJGrnxoA\nAABYOctEhkeSHK+qm6vq+iS3J1n/7pPd/Vx3H+7um7r7piQPJznV3Ru7MjEAAACwknaMDN39QpK7\nkzyY5KtJHujux6rq3qo6tdsDAgAAAAfDoWUWdfe5JOcuOvf+y6x9y4sfCwAAADhoXvSNHwEAAAAS\nkQEAAAAYIjIAAAAAI0QGAAAAYITIAAAAAIwQGQAAAIARIgMAAAAwQmQAAAAARogMAAAAwAiRAQAA\nABghMgAAAAAjRAYAAABghMgAAAAAjBAZAAAAgBEiAwAAADBCZAAAAABGiAwAAADACJEBAAAAGCEy\nAAAAACNEBgAAAGCEyAAAAACMEBkAAACAESIDAAAAMEJkAAAAAEaIDAAAAMAIkQEAAAAYITIAAAAA\nI0QGAAAAYITIAAAAAIwQGQAAAIARIgMAAAAwQmQAAAAARogMAAAAwAiRAQAAABghMgAAAAAjRAYA\nAABghMgAAAAAjBAZAAAAgBEiAwAAADBCZAAAAABGiAwAAADACJEBAAAAGCEyAAAAACNEBgAAAGCE\nyAAAAACMEBkAAACAEUtFhqo6WVVPVNVmVd1zieffU1WPV9WjVfX3VfXq+VEBAACAVbZjZKiq65Lc\nl+TWJCeS3FFVJy5a9uUka939c0k+m+Qj04MCAAAAq22ZKxluSbLZ3U929/NJ7k9yevuC7n6ou7+1\nOHw4ydHZMQEAAIBVt0xkuCHJU9uOzy/OXc5dSf7uUk9U1Zmq2qiqjQsXLiw/JQAAALDyRm/8WFV3\nJllL8tFLPd/dZ7t7rbvXjhw5MvnRAAAAwD47tMSap5Mc23Z8dHHue1TV25L8XpJf7O7vzIwHAAAA\nHBTLXMnwSJLjVXVzVV2f5PYk69sXVNUbkvxZklPd/cz8mAAAAMCq2zEydPcLSe5O8mCSryZ5oLsf\nq6p7q+rUYtlHk/xYkr+uqn+qqvXLvB0AAADwQ2qZr0uku88lOXfRufdve/y24bkAAACAA2b0xo8A\nAADAtUtkAAAAAEaIDAAAAMAIkQEAAAAYITIAAAAAI0QGAAAAYITIAAAAAIwQGQAAAIARIgMAAAAw\nQmQAAAAARogMAAAAwAiRAQAAABghMgAAAAAjRAYAAABghMgAAAAAjBAZAAAAgBEiAwAAADBCZAAA\nAABGiAwAAADACJEBAAAAGCEyAAAAACNEBgAAAGCEyAAAAACMEBkAAACAESIDAAAAMEJkAAAAAEaI\nDAAAAMAIkQEAAAAYITIAAAAAI0QGAAAAYITIAAAAAIwQGQAAAIARIgMAAAAwQmQAAAAARogMAAAA\nwAiRAQAAABghMgAAAAAjRAYAAABghMgAAAAAjBAZAAAAgBEiAwAAADBCZAAAAABGiAwAAADACJEB\nAAAAGCEyAAAAACOWigxVdbKqnqiqzaq65xLPv7Sq/mrx/Ber6qbpQQEAAIDVtmNkqKrrktyX5NYk\nJ5LcUVUnLlp2V5Jnu/unkvxhkg9PDwoAAACstmWuZLglyWZ3P9ndzye5P8npi9acTvKpxePPJnlr\nVdXcmAAAAMCqWyYy3JDkqW3H5xfnLrmmu19I8lySV00MCAAAABwMh/byw6rqTJIzi8PvVNVX9vLz\n4YA5nOTr+z0ErDB7BK7MHoErs0fg8n76al+4TGR4OsmxbcdHF+cuteZ8VR1K8sok37j4jbr7bJKz\nSVJVG929djVDw7XAHoErs0fgyuwRuDJ7BC6vqjau9rXLfF3ikSTHq+rmqro+ye1J1i9as57kHYvH\nv5rkH7q7r3YoAAAA4ODZ8UqG7n6hqu5O8mCS65J8orsfq6p7k2x093qSjyf5dFVtJvmvbIUIAAAA\n4Bqy1D0ZuvtcknMXnXv/tsffTvJrP+Bnn/0B18O1xh6BK7NH4MrsEbgyewQu76r3R/lWAwAAADBh\nmXsyAAAAAOxo1yNDVZ2sqieqarOq7rnE8y+tqr9aPP/Fqrppt2eCVbLEHnlPVT1eVY9W1d9X1av3\nY07YLzvtkW3rfqWquqrcKZxrxjL7o6p+ffFz5LGq+su9nhH20xJ/z7qxqh6qqi8v/q51237MCful\nqj5RVc9U1Vcu83xV1R8t9tCjVfXGnd5zVyNDVV2X5L4ktyY5keSOqjpx0bK7kjzb3T+V5A+TfHg3\nZ4JVsuQe+XKSte7+uSSfTfKRvZ0S9s+SeyRV9fIkv5Xki3s7IeyfZfZHVR1P8r4kb+7un0ny23s+\nKOyTJX+G/H6SB7r7Ddm6ef0f7+2UsO8+meTkFZ6/NcnxxZ8zSf5kpzfc7SsZbkmy2d1PdvfzSe5P\ncvqiNaeTfGrx+LNJ3lpVtctzwarYcY9090Pd/a3F4cNJju7xjLCflvk5kiQfylak/vZeDgf7bJn9\n8e4k93X3s0nS3c/s8Yywn5bZI53kFYvHr0zyH3s4H+y77v5ctn5D5OWcTvIXveXhJD9eVT95pffc\n7chwQ5Knth2fX5y75JrufiHJc0letctzwapYZo9sd1eSv9vViWC17LhHFpftHevuv93LwWAFLPMz\n5DVJXlNVn6+qh6vqSv9bBT9sltkjH0xyZ1Wdz9Zv0/vNvRkNDowf9N8ry/0KS2D/VdWdSdaS/OJ+\nzwKroqpekuRjSd65z6PAqjqUrUtc35KtK+E+V1U/293f3NepYHXckeST3f0HVfULST5dVa/v7v/d\n78HgoNrtKxmeTnJs2/HRxblLrqmqQ9m6TOkbuzwXrIpl9kiq6m1Jfi/Jqe7+zh7NBqtgpz3y8iSv\nT/KPVfVvSd6UZN3NH7lGLPMz5HyS9e7+7+7+1yT/kq3oANeCZfbIXUkeSJLu/kKSH01yeE+mg4Nh\nqX+vbLfbkeGRJMer6uaquj5bN1NZv2jNepJ3LB7/apJ/6O7e5blgVey4R6rqDUn+LFuBwXdpudZc\ncY9093Pdfbi7b+rum7J135JT3b2xP+PCnlrm71l/k62rGFJVh7P19Ykn93JI2EfL7JGvJXlrklTV\n67IVGS7s6ZSw2taTvH3xWybelOS57v7PK71gV78u0d0vVNXdSR5Mcl2ST3T3Y1V1b5KN7l5P8vFs\nXZa0ma0bTty+mzPBKllyj3w0yY8l+evFPVG/1t2n9m1o2ENL7hG4Ji25Px5M8stV9XiS/0nyu93t\nilGuCUvukfcm+fOq+p1s3QTynf7Dk2tJVX0mWzH68OLeJB9I8iNJ0t1/mq17ldyWZDPJt5K8a8f3\ntIcAAACACbv9dQkAAADgGiEyAAAAACNEBgAAAGCEyAAAAACMEBkAAACAESIDAAAAMEJkAAAAAEaI\nDAAAAMCI/wNg17DNza1rEQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x10b5e4860>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# I don't like this Pie Chart\n",
    "fig1, ax1 = plt.subplots()\n",
    "fig1.set_size_inches(18, 5)\n",
    "ax1.bar(labels, values, 1/1.5, color=\"blue\")\n",
    "#ax1.pie(values, labels=labels, shadow=True, startangle=90)\n",
    "#ax1.axis('equal')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a report on metadata\n",
    "columns = ['PidEntryName',\n",
    "           'PidEntryPrefix',\n",
    "           'PidEntryUrl', \n",
    "           'ResourceInfo', \n",
    "           'ResourceInstitution', \n",
    "           'ResourceLocation', \n",
    "           'ResourceOfficial', \n",
    "           'ResourcePrefix', \n",
    "           'ResourceLocalId', \n",
    "           'WasMetadataFound', \n",
    "           'MetadataContent', \n",
    "           'ResourceTestUrl', \n",
    "           'MetadataServiceResponseStatus', \n",
    "           'MetadataServiceResponseError']\n",
    "metadata_report = pandas.DataFrame(columns=columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Prepare the URLs and initial report (I could have done everything in one pass, but this is just investigating the dataset)\n",
    "for pid_entry in resolver_dump:\n",
    "    entry = pandas.Series(['---'] * len(columns), index=columns)\n",
    "    entry.PidEntryName = pid_entry['name']\n",
    "    entry.PidEntryPrefix = pid_entry['prefix']\n",
    "    entry.PidEntryUrl = pid_entry['url']\n",
    "    if ('resources' not in pid_entry) or (not pid_entry['resources']):\n",
    "        metadata_report = metadata_report.append(entry, ignore_index=True)\n",
    "    for resource in pid_entry['resources']:\n",
    "        resource_entry = entry.copy()\n",
    "        resource_entry.ResourceInfo = resource['info']\n",
    "        resource_entry.ResourceInstitution = resource.get('institution', '---')\n",
    "        resource_entry.ResourceLocation = resource.get('location', '---')\n",
    "        resource_entry.ResourceOfficial = resource.get('official', '---')\n",
    "        resource_entry.ResourcePrefix = resource.get('resourcePrefix', '---')\n",
    "        resource_entry.ResourceLocalId = resource.get('localId', '---')\n",
    "        resource_entry.WasMetadataFound = 'No'\n",
    "        resource_entry.ResourceTestUrl = None\n",
    "        if ('accessURL' in resource) and ('localId' in resource):\n",
    "            resource_entry.ResourceTestUrl = resource['accessURL'].replace('{$id}', resource['localId'])\n",
    "        metadata_report = metadata_report.append(resource_entry, ignore_index=True)        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count                                     761\n",
       "unique                                    760\n",
       "top       http://zfin.org/ZDB-GENO-980202-899\n",
       "freq                                        2\n",
       "Name: ResourceTestUrl, dtype: object"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_report.ResourceTestUrl.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PidEntryName</th>\n",
       "      <th>PidEntryPrefix</th>\n",
       "      <th>PidEntryUrl</th>\n",
       "      <th>ResourceInfo</th>\n",
       "      <th>ResourceInstitution</th>\n",
       "      <th>ResourceLocation</th>\n",
       "      <th>ResourceOfficial</th>\n",
       "      <th>ResourcePrefix</th>\n",
       "      <th>ResourceLocalId</th>\n",
       "      <th>WasMetadataFound</th>\n",
       "      <th>MetadataContent</th>\n",
       "      <th>ResourceTestUrl</th>\n",
       "      <th>MetadataServiceResponseStatus</th>\n",
       "      <th>MetadataServiceResponseError</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ChEBI</td>\n",
       "      <td>chebi</td>\n",
       "      <td>http://identifiers.org/chebi</td>\n",
       "      <td>ChEBI (Chemical Entities of Biological Interest)</td>\n",
       "      <td>European Bioinformatics Institute, Hinxton, Ca...</td>\n",
       "      <td>UK</td>\n",
       "      <td>True</td>\n",
       "      <td>ebi</td>\n",
       "      <td>36927</td>\n",
       "      <td>No</td>\n",
       "      <td>---</td>\n",
       "      <td>http://www.ebi.ac.uk/chebi/searchId.do?chebiId...</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ChEBI</td>\n",
       "      <td>chebi</td>\n",
       "      <td>http://identifiers.org/chebi</td>\n",
       "      <td>ChEBI through OLS</td>\n",
       "      <td>European Bioinformatics Institute, Hinxton, Ca...</td>\n",
       "      <td>UK</td>\n",
       "      <td>False</td>\n",
       "      <td>ols</td>\n",
       "      <td>36927</td>\n",
       "      <td>No</td>\n",
       "      <td>---</td>\n",
       "      <td>http://www.ebi.ac.uk/ols/ontologies/chebi/term...</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ChEBI</td>\n",
       "      <td>chebi</td>\n",
       "      <td>http://identifiers.org/chebi</td>\n",
       "      <td>ChEBI through BioPortal</td>\n",
       "      <td>National Center for Biomedical Ontology, Stanford</td>\n",
       "      <td>USA</td>\n",
       "      <td>False</td>\n",
       "      <td>bptl</td>\n",
       "      <td>36927</td>\n",
       "      <td>No</td>\n",
       "      <td>---</td>\n",
       "      <td>http://purl.bioontology.org/ontology/CHEBI/CHE...</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Ensembl</td>\n",
       "      <td>ensembl</td>\n",
       "      <td>http://identifiers.org/ensembl</td>\n",
       "      <td>Ensembl at Sanger/EMBL-EBI</td>\n",
       "      <td>Sanger Institute and European Bioinformatics I...</td>\n",
       "      <td>UK</td>\n",
       "      <td>True</td>\n",
       "      <td>ebi</td>\n",
       "      <td>ENSG00000139618</td>\n",
       "      <td>No</td>\n",
       "      <td>---</td>\n",
       "      <td>http://www.ensembl.org/id/ENSG00000139618</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Ensembl</td>\n",
       "      <td>ensembl</td>\n",
       "      <td>http://identifiers.org/ensembl</td>\n",
       "      <td>Ensembl US West mirror</td>\n",
       "      <td>---</td>\n",
       "      <td>USA</td>\n",
       "      <td>False</td>\n",
       "      <td>---</td>\n",
       "      <td>ENSG00000139618</td>\n",
       "      <td>No</td>\n",
       "      <td>---</td>\n",
       "      <td>http://uswest.ensembl.org/id/ENSG00000139618</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  PidEntryName PidEntryPrefix                     PidEntryUrl  \\\n",
       "0        ChEBI          chebi    http://identifiers.org/chebi   \n",
       "1        ChEBI          chebi    http://identifiers.org/chebi   \n",
       "2        ChEBI          chebi    http://identifiers.org/chebi   \n",
       "3      Ensembl        ensembl  http://identifiers.org/ensembl   \n",
       "4      Ensembl        ensembl  http://identifiers.org/ensembl   \n",
       "\n",
       "                                       ResourceInfo  \\\n",
       "0  ChEBI (Chemical Entities of Biological Interest)   \n",
       "1                                 ChEBI through OLS   \n",
       "2                           ChEBI through BioPortal   \n",
       "3                        Ensembl at Sanger/EMBL-EBI   \n",
       "4                            Ensembl US West mirror   \n",
       "\n",
       "                                 ResourceInstitution ResourceLocation  \\\n",
       "0  European Bioinformatics Institute, Hinxton, Ca...               UK   \n",
       "1  European Bioinformatics Institute, Hinxton, Ca...               UK   \n",
       "2  National Center for Biomedical Ontology, Stanford              USA   \n",
       "3  Sanger Institute and European Bioinformatics I...               UK   \n",
       "4                                                ---              USA   \n",
       "\n",
       "  ResourceOfficial ResourcePrefix  ResourceLocalId WasMetadataFound  \\\n",
       "0             True            ebi            36927               No   \n",
       "1            False            ols            36927               No   \n",
       "2            False           bptl            36927               No   \n",
       "3             True            ebi  ENSG00000139618               No   \n",
       "4            False            ---  ENSG00000139618               No   \n",
       "\n",
       "  MetadataContent                                    ResourceTestUrl  \\\n",
       "0             ---  http://www.ebi.ac.uk/chebi/searchId.do?chebiId...   \n",
       "1             ---  http://www.ebi.ac.uk/ols/ontologies/chebi/term...   \n",
       "2             ---  http://purl.bioontology.org/ontology/CHEBI/CHE...   \n",
       "3             ---          http://www.ensembl.org/id/ENSG00000139618   \n",
       "4             ---       http://uswest.ensembl.org/id/ENSG00000139618   \n",
       "\n",
       "  MetadataServiceResponseStatus MetadataServiceResponseError  \n",
       "0                           ---                          ---  \n",
       "1                           ---                          ---  \n",
       "2                           ---                          ---  \n",
       "3                           ---                          ---  \n",
       "4                           ---                          ---  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_report.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parallel call to metadata service, this is a lot slower, good enough as proof of concept\n",
    "#metadata_requests = {index: threading.Thread(target=get_metadata_for_url(metadata_report.loc[index].ResourceTestUrl)) for index in range(metadata_report.shape[0]) if metadata_report.loc[index].ResourceTestUrl}\n",
    "\n",
    "# Parallel wrapper - Version using multiprocessing, it crashes within Jupyter\n",
    "#def metadata_request_parallel_wrapper(context, url):\n",
    "#    context.put(get_metadata_for_url(url))\n",
    "#metadata_requests = {index: multiprocessing.Process(target=metadata_request_parallel_wrapper, args=(multiprocessing.Queue(), metadata_report.loc[index].ResourceTestUrl),) for index in range(metadata_report.shape[0]) if metadata_report.loc[index].ResourceTestUrl}\n",
    "#metadata_requests = {index: multiprocessing.Process(target=get_metadata_for_url, args=(metadata_report.loc[index].ResourceTestUrl,)) for index in range(metadata_report.shape[0]) if metadata_report.loc[index].ResourceTestUrl}\n",
    "#[process.start() for process in metadata_requests.values()]\n",
    "\n",
    "# Another approach, with Thread Pool\n",
    "pool = ThreadPool(processes=multiprocessing.cpu_count() * 4)\n",
    "metadata_requests = {index: pool.apply_async(get_metadata_for_url, (metadata_report.loc[index].ResourceTestUrl)) \n",
    "                     for index in range(metadata_report.shape[0]) \n",
    "                     if metadata_report.loc[index].ResourceTestUrl}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "761"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(metadata_requests)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pool."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
