{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# This script provides some reporting on the resolver data from identifiers.org\n",
    "#\n",
    "# Author: Manuel Bernal Llinares <mbdebian@gmail.com>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import time\n",
    "import random\n",
    "import requests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Endpoint from where the information is coming\n",
    "identifiersorg_resolver_data_url = \"https://identifiers.org/rest/collections/expand\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize pseudo-random number generator\n",
    "random.seed(time.time())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Request Helper\n",
    "def make_rest_request_content_type_json(url):\n",
    "    # TODO - Magic number here!!!\n",
    "    n_attempts = 42\n",
    "    response = None\n",
    "    while n_attempts:\n",
    "        n_attempts -= 1\n",
    "        try:\n",
    "            response = requests.get(url, headers={\"Content-Type\": \"application/json\"})\n",
    "        except Exception as e:\n",
    "            # Any possible exception counts towards the attempt counter\n",
    "            # Random wait - TODO - Another magic number!!!\n",
    "            time.sleep(random.randint(30))\n",
    "            continue\n",
    "        if response.ok:\n",
    "            return response.json()\n",
    "        # Random wait - TODO - Another magic number!!!\n",
    "        time.sleep(random.randint(10))\n",
    "    response.raise_for_status()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the resolver data\n",
    "resolver_dump = make_rest_request_content_type_json(identifiersorg_resolver_data_url)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Workout how many prefixes there are in identifiers.org\n",
    "prefixes = [pid_entry['prefix'] for pid_entry in resolver_dump]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['chebi',\n",
       " 'ensembl',\n",
       " 'ec-code',\n",
       " 'uniprot',\n",
       " 'taxonomy',\n",
       " 'biomodels.db',\n",
       " 'miriam.collection',\n",
       " 'icd',\n",
       " 'intact',\n",
       " 'interpro',\n",
       " 'kegg.pathway',\n",
       " 'kegg.compound',\n",
       " 'kegg.reaction',\n",
       " 'pubmed',\n",
       " 'omim',\n",
       " 'pirsf',\n",
       " 'reactome',\n",
       " 'doi',\n",
       " 'pdb',\n",
       " 'go',\n",
       " 'sgd',\n",
       " 'sbo',\n",
       " 'kegg.drug',\n",
       " 'kegg.glycan',\n",
       " 'wormbase',\n",
       " 'pfam',\n",
       " 'insdc',\n",
       " 'flybase',\n",
       " 'wormpep',\n",
       " 'prosite',\n",
       " 'pubchem.substance',\n",
       " 'pubchem.compound',\n",
       " 'arxiv',\n",
       " 'arrayexpress',\n",
       " 'mgi',\n",
       " 'sabiork.reaction',\n",
       " 'refseq',\n",
       " 'tcdb',\n",
       " 'uniparc',\n",
       " 'mint',\n",
       " 'dip',\n",
       " 'signaling-gateway',\n",
       " 'resid',\n",
       " 'rgd',\n",
       " 'tair.protein',\n",
       " 'tair.gene',\n",
       " 'tair.locus',\n",
       " 'hmdb',\n",
       " 'lipidmaps',\n",
       " 'peptideatlas',\n",
       " 'geo',\n",
       " 'eco',\n",
       " 'mod',\n",
       " 'sgd.pathways',\n",
       " 'biogrid',\n",
       " 'merops',\n",
       " 'panther.family',\n",
       " 'prints',\n",
       " 'ligandexpo',\n",
       " 'aclame',\n",
       " 'isbn',\n",
       " 'pride',\n",
       " '3dmet',\n",
       " 'fma',\n",
       " 'matrixdb.association',\n",
       " 'ncbigene',\n",
       " 'kegg.genes',\n",
       " 'brenda',\n",
       " 'pubchem.bioassay',\n",
       " 'pathwaycommons',\n",
       " 'hovergen',\n",
       " 'mmmp:biomaps',\n",
       " 'wikipathways',\n",
       " 'macie',\n",
       " 'mirbase',\n",
       " 'zfin',\n",
       " 'hgnc',\n",
       " 'so',\n",
       " 'rhea',\n",
       " 'unipathway',\n",
       " 'chembl.compound',\n",
       " 'chembl.target',\n",
       " 'sabiork.kineticrecord',\n",
       " 'lgic',\n",
       " 'atc',\n",
       " 'pharmgkb.pathways',\n",
       " 'pharmgkb.disease',\n",
       " 'pharmgkb.drug',\n",
       " 'ttd.drug',\n",
       " 'ttd.target',\n",
       " 'neurondb',\n",
       " 'neuromorpho',\n",
       " 'chemidplus',\n",
       " 'biosystems',\n",
       " 'ctd.chemical',\n",
       " 'ctd.disease',\n",
       " 'ctd.gene',\n",
       " 'bionumbers',\n",
       " 'drugbank',\n",
       " 't3db',\n",
       " 'smpdb',\n",
       " 'phosphosite.protein',\n",
       " 'genedb',\n",
       " 'biomodels.teddy',\n",
       " 'biomodels.kisao',\n",
       " 'psimi',\n",
       " 'cl',\n",
       " 'bto',\n",
       " 'pato',\n",
       " 'pdb-ccd',\n",
       " 'glycomedb',\n",
       " 'lipidbank',\n",
       " 'kegg.orthology',\n",
       " 'prodom',\n",
       " 'smart',\n",
       " 'cdd',\n",
       " 'ro',\n",
       " 'mmdb',\n",
       " 'imex',\n",
       " 'irefweb',\n",
       " 'mpid',\n",
       " 'phosphosite.residue',\n",
       " 'neurolex',\n",
       " 'obi',\n",
       " 'sabiork.ec',\n",
       " 'opb',\n",
       " 'jws',\n",
       " 'modeldb',\n",
       " 'subtiwiki',\n",
       " 'pid.pathway',\n",
       " 'doqcs.model',\n",
       " 'doqcs.pathway',\n",
       " 'uo',\n",
       " 'clinicaltrials',\n",
       " 'chemspider',\n",
       " 'ncit',\n",
       " 'biocatalogue.service',\n",
       " 'pr',\n",
       " 'omia',\n",
       " 'chembank',\n",
       " 'csa',\n",
       " 'cgd',\n",
       " 'antweb',\n",
       " 'pmc',\n",
       " 'amoebadb',\n",
       " 'cryptodb',\n",
       " 'plasmodb',\n",
       " 'giardiadb',\n",
       " 'microsporidia',\n",
       " 'toxoplasma',\n",
       " 'trichdb',\n",
       " 'tritrypdb',\n",
       " 'bdgp.insertion',\n",
       " 'beetlebase',\n",
       " 'bold.taxonomy',\n",
       " 'dbest',\n",
       " 'dbprobe',\n",
       " 'dbsnp',\n",
       " 'unists',\n",
       " 'ecogene',\n",
       " 'gabi',\n",
       " 'greengenes',\n",
       " 'grin.taxonomy',\n",
       " 'hinv.locus',\n",
       " 'hinv.transcript',\n",
       " 'hinv.protein',\n",
       " 'homd.seq',\n",
       " 'homd.taxon',\n",
       " 'ird.segment',\n",
       " 'isfinder',\n",
       " 'jcm',\n",
       " 'img.taxon',\n",
       " 'img.gene',\n",
       " 'maizegdb.locus',\n",
       " 'mycobank',\n",
       " 'nbrc',\n",
       " 'pseudomonas',\n",
       " 'gramene.protein',\n",
       " 'gramene.gene',\n",
       " 'gramene.taxonomy',\n",
       " 'gramene.qtl',\n",
       " 'sgn',\n",
       " 'xenbase',\n",
       " 'bioportal',\n",
       " 'miriam.resource',\n",
       " 'edam',\n",
       " 'pmdb',\n",
       " 'arachnoserver',\n",
       " 'biocyc',\n",
       " 'cazy',\n",
       " 'goa',\n",
       " 'paleodb',\n",
       " 'compulyeast',\n",
       " 'disprot',\n",
       " 'echobase',\n",
       " 'eggnog',\n",
       " 'ensembl.bacteria',\n",
       " 'ensembl.protist',\n",
       " 'ensembl.metazoa',\n",
       " 'ensembl.plant',\n",
       " 'ensembl.fungi',\n",
       " 'hcvdb',\n",
       " 'genatlas',\n",
       " 'cath.superfamily',\n",
       " 'cath.domain',\n",
       " 'genefarm',\n",
       " 'gpcrdb',\n",
       " 'hogenom',\n",
       " 'genetree',\n",
       " 'hssp',\n",
       " 'myco.tuber',\n",
       " 'myco.lepra',\n",
       " 'myco.marinum',\n",
       " 'myco.smeg',\n",
       " 'orphanet',\n",
       " 'orthodb',\n",
       " 'peroxibase',\n",
       " 'phylomedb',\n",
       " 'pmap.substratedb',\n",
       " 'pmap.cutdb',\n",
       " 'protclustdb',\n",
       " 'pmp',\n",
       " 'protonet.proteincard',\n",
       " 'protonet.cluster',\n",
       " 'rebase',\n",
       " 'swiss-model',\n",
       " 'vectorbase',\n",
       " 'doid',\n",
       " 'cco',\n",
       " 'mirbase.mature',\n",
       " 'nextprot',\n",
       " 'cas',\n",
       " 'kegg.genome',\n",
       " 'kegg.metagenome',\n",
       " 'narcis',\n",
       " 'jcsd',\n",
       " 'pw',\n",
       " 'insdc.sra',\n",
       " 'scretf',\n",
       " 'pharmgkb.gene',\n",
       " 'mirnest',\n",
       " 'napp',\n",
       " 'noncodev3',\n",
       " 'virsirna',\n",
       " 'elm',\n",
       " 'mimodb',\n",
       " 'sitex',\n",
       " 'bykdb',\n",
       " 'conoserver',\n",
       " 'topfind',\n",
       " 'mipmod',\n",
       " 'cellimage',\n",
       " 'combine.specifications',\n",
       " 'zfin.phenotype',\n",
       " 'zfin.expression',\n",
       " 'cabri',\n",
       " 'cygd',\n",
       " 'huge',\n",
       " 'bindingDB',\n",
       " 'string',\n",
       " 'stitch',\n",
       " 'atcvet',\n",
       " 'phenolexplorer',\n",
       " 'snomedct',\n",
       " 'mesh.2012',\n",
       " 'knapsack',\n",
       " 'cdpd',\n",
       " 'massbank',\n",
       " 'gmd',\n",
       " 'homologene',\n",
       " 'umbbd.compound',\n",
       " 'abs',\n",
       " 'apd',\n",
       " 'chemdb',\n",
       " 'dpv',\n",
       " 'iuphar.receptor',\n",
       " 'aceview.worm',\n",
       " 'asap',\n",
       " 'atcc',\n",
       " 'bdgp.est',\n",
       " 'dictybase.gene',\n",
       " 'imgt.ligm',\n",
       " 'worfdb',\n",
       " 'nextdb',\n",
       " 'soybase',\n",
       " 'hamap',\n",
       " 'rouge',\n",
       " 'arrayexpress.platform',\n",
       " 'cgsc',\n",
       " 'dragondb.dna',\n",
       " 'dragondb.protein',\n",
       " 'dragondb.locus',\n",
       " 'dragondb.allele',\n",
       " 'issn',\n",
       " 'merops.family',\n",
       " 'mo',\n",
       " 'nasc',\n",
       " 'niaest',\n",
       " 'pazar',\n",
       " 'po',\n",
       " 'rnamods',\n",
       " 'treebase',\n",
       " 'tgd',\n",
       " 'tigrfam',\n",
       " 'atfdb.family',\n",
       " 'iuphar.family',\n",
       " 'dbg2introns',\n",
       " 'sdbs',\n",
       " 'vbase2',\n",
       " 'spike.map',\n",
       " 'metlin',\n",
       " 'genecards',\n",
       " 'mmrrc',\n",
       " 'umbbd.reaction',\n",
       " 'umbbd.enzyme',\n",
       " 'umbbd.pathway',\n",
       " 'umbbd.rule',\n",
       " 'mirex',\n",
       " 'dictybase.est',\n",
       " 'imgt.hla',\n",
       " 'flystock',\n",
       " 'opm',\n",
       " 'allergome',\n",
       " 'pombase',\n",
       " 'hpa',\n",
       " 'jaxmice',\n",
       " 'ubio.namebank',\n",
       " 'yetfasco',\n",
       " 'tarbase',\n",
       " 'charprot',\n",
       " 'oma.protein',\n",
       " 'oma.grp',\n",
       " 'ncbiprotein',\n",
       " 'genpept',\n",
       " 'unigene',\n",
       " 'bitterdb.rec',\n",
       " 'bitterdb.cpd',\n",
       " 'bioproject',\n",
       " 'biosample',\n",
       " 'piroplasma',\n",
       " 'unite',\n",
       " 'ncim',\n",
       " 'proglyc',\n",
       " 'polbase',\n",
       " 'nuclearbd',\n",
       " 'supfam',\n",
       " 'ricegap',\n",
       " 'pina',\n",
       " 'tissuelist',\n",
       " 'bacmap.biog',\n",
       " 'hgnc.symbol',\n",
       " 'panther.pathway',\n",
       " 'fairsharing',\n",
       " 'fungidb',\n",
       " 'darc',\n",
       " 'drsc',\n",
       " 'oridb.schizo',\n",
       " 'oridb.sacch',\n",
       " 'pscdb',\n",
       " 'scop',\n",
       " 'ena.embl',\n",
       " 'dommino',\n",
       " 'panther.node',\n",
       " 'ccds',\n",
       " 'lrg',\n",
       " 'hprd',\n",
       " 'gxa.gene',\n",
       " 'gxa.expt',\n",
       " 'metabolights',\n",
       " 'nbn',\n",
       " 'orcid',\n",
       " 'inchi',\n",
       " 'wikipedia.en',\n",
       " 'phosphopoint.kinase',\n",
       " 'phosphopoint.protein',\n",
       " 'inchikey',\n",
       " 'uniprot.isoform',\n",
       " 'kegg.environ',\n",
       " 'cldb',\n",
       " 'efo',\n",
       " 'hgmd',\n",
       " 'aphidbase.transcript',\n",
       " 'affy.probeset',\n",
       " 'treefam',\n",
       " 'caps',\n",
       " 'cubedb',\n",
       " 'ideal',\n",
       " 'stap',\n",
       " 'pocketome',\n",
       " 'gold.genome',\n",
       " 'gold.meta',\n",
       " 'bugbase.protocol',\n",
       " 'bugbase.expt',\n",
       " 'tol',\n",
       " 'vario',\n",
       " 'vipr',\n",
       " 'epd',\n",
       " 'rfam',\n",
       " 'fbol',\n",
       " 'aftol.taxonomy',\n",
       " 'aspgd.locus',\n",
       " 'aspgd.protein',\n",
       " 'autdb',\n",
       " 'bacmap.map',\n",
       " 'bgee.family',\n",
       " 'bgee.gene',\n",
       " 'bgee.stage',\n",
       " 'bgee.organ',\n",
       " 'biocarta.pathway',\n",
       " 'panther.pthcmp',\n",
       " 'gmd.profile',\n",
       " 'gmd.gcms',\n",
       " 'gmd.ref',\n",
       " 'gmd.analyte',\n",
       " 'intact.molecule',\n",
       " 'depod',\n",
       " 'cst',\n",
       " 'cst.ab',\n",
       " 'ndc',\n",
       " 'phytozome.locus',\n",
       " 'subtilist',\n",
       " 'dailymed',\n",
       " 'sider.drug',\n",
       " 'sider.effect',\n",
       " 'wikigenes',\n",
       " 'broad',\n",
       " 'coriell',\n",
       " 'corum',\n",
       " 'ecoliwiki',\n",
       " 'jstor',\n",
       " 'ma',\n",
       " 'uberon',\n",
       " 'unimod',\n",
       " 'vbrc',\n",
       " 'viralzone',\n",
       " 'go.ref',\n",
       " 'rgd.qtl',\n",
       " 'rgd.strain',\n",
       " 'door',\n",
       " 'degradome',\n",
       " 'dbd',\n",
       " 'datf',\n",
       " 'iuphar.ligand',\n",
       " 'molbase',\n",
       " 'yrcpdr',\n",
       " 'yid',\n",
       " 'funcbase.fly',\n",
       " 'funcbase.human',\n",
       " 'funcbase.mouse',\n",
       " 'funcbase.yeast',\n",
       " 'ydpm',\n",
       " 'wormbase.rnai',\n",
       " 'funcat',\n",
       " 'pass2',\n",
       " 'iceberg.element',\n",
       " 'iceberg.family',\n",
       " 'vfdb.genus',\n",
       " 'vfdb.gene',\n",
       " 'mesh.2013',\n",
       " 'kegg.module',\n",
       " 'kegg.disease',\n",
       " 'medlineplus',\n",
       " 'ligandbox',\n",
       " 'glycoepitope',\n",
       " 'jcggdb',\n",
       " 'noncodev4.gene',\n",
       " 'noncodev4.rna',\n",
       " 'oryzabase.gene',\n",
       " 'oryzabase.mutant',\n",
       " 'oryzabase.strain',\n",
       " 'oryzabase.stage',\n",
       " 'otl',\n",
       " 'genewiki',\n",
       " 'paxdb.organism',\n",
       " 'paxdb.protein',\n",
       " 'pdb.ligand',\n",
       " 'merops.inhibitor',\n",
       " 'adw',\n",
       " 'glida.gpcr',\n",
       " 'glida.ligand',\n",
       " 'grsdb',\n",
       " 'fsnp',\n",
       " 'hdr',\n",
       " 'norine',\n",
       " 'ordb',\n",
       " 'odor',\n",
       " 'p3db.protein',\n",
       " 'p3db.site',\n",
       " 'topdb',\n",
       " 'cattleqtldb',\n",
       " 'chickenqtldb',\n",
       " 'pigqtldb',\n",
       " 'sheepqtldb',\n",
       " 'gramene.growthstage',\n",
       " 'ncbigi',\n",
       " 'ebimetagenomics.samp',\n",
       " 'ega.study',\n",
       " 'ega.dataset',\n",
       " 'px',\n",
       " 'biomodels.vocabulary',\n",
       " 'pride.project',\n",
       " 'antibodyregistry',\n",
       " 'mamo',\n",
       " 'eo',\n",
       " 'idot',\n",
       " 'hgnc.family',\n",
       " 'yeastintron',\n",
       " 'ardb',\n",
       " 'proteomicsdb.protein',\n",
       " 'proteomicsdb.peptide',\n",
       " 'hpm.protein',\n",
       " 'hpm.peptide',\n",
       " 'drugbankv4.target',\n",
       " 'zinc',\n",
       " 'foodb.compound',\n",
       " 'unii',\n",
       " 'orphanet.ordo',\n",
       " 'psipar',\n",
       " 'clinvar.record',\n",
       " 'ebimetagenomics.proj',\n",
       " 'euclinicaltrials',\n",
       " 'google.patent',\n",
       " 'uspto',\n",
       " 'cpc',\n",
       " 'gwascentral.study',\n",
       " 'exac.variant',\n",
       " 'gwascentral.marker',\n",
       " 'gwascentral.phenotype',\n",
       " 'lincs.cell',\n",
       " 'lincs.protein',\n",
       " 'lincs.smallmolecule',\n",
       " 'exac.transcript',\n",
       " 'exac.gene',\n",
       " 'wikidata',\n",
       " 'swisslipid',\n",
       " 'unipathway.compound',\n",
       " 'seed',\n",
       " 'seed.compound',\n",
       " 'bigg.model',\n",
       " 'bigg.compartment',\n",
       " 'bigg.metabolite',\n",
       " 'bigg.reaction',\n",
       " 'rrid',\n",
       " 'umls',\n",
       " 'mesh',\n",
       " 'emdb',\n",
       " 'mirtarbase',\n",
       " 'meddra',\n",
       " 'dashr',\n",
       " 'dashr.expression',\n",
       " 'splash',\n",
       " 'metanetx.chemical',\n",
       " 'metanetx.reaction',\n",
       " 'metanetx.compartment',\n",
       " 'unipathway.reaction',\n",
       " 'hp',\n",
       " 'sasbdb',\n",
       " 'hgnc.genefamily',\n",
       " 'mdm',\n",
       " 'probonto',\n",
       " 'apid.interactions',\n",
       " 'storedb',\n",
       " 'kegg',\n",
       " 'planttfdb',\n",
       " 'mp',\n",
       " 'worms',\n",
       " 'multicellds.collection',\n",
       " 'multicellds.snapshot',\n",
       " 'multicellds.cell_line',\n",
       " 'ecyano.entity',\n",
       " 'ecyano.model',\n",
       " 'ecyano.rule',\n",
       " 'lincs.data',\n",
       " 'agricola',\n",
       " 'ms',\n",
       " 'envo',\n",
       " 'ark',\n",
       " 'grid',\n",
       " 'medgen',\n",
       " 'clinvar.submission',\n",
       " 'clinvar',\n",
       " 'bao',\n",
       " 'cellosaurus',\n",
       " 'mir',\n",
       " 'pgx',\n",
       " 'cosmic',\n",
       " 'dbgap',\n",
       " 'iao',\n",
       " 'gdc',\n",
       " 'omit',\n",
       " 'insdc.cds',\n",
       " 'insdc.gca',\n",
       " 'glytoucan',\n",
       " 'metacyc.compound',\n",
       " 'metacyc.reaction',\n",
       " 'd1id',\n",
       " 'cath',\n",
       " 'mmp.ref',\n",
       " 'mmp.db',\n",
       " 'mmp.cat',\n",
       " 'biotools',\n",
       " 'aop',\n",
       " 'ngl',\n",
       " 'erv',\n",
       " 'napdi',\n",
       " 'arraymap',\n",
       " 'begdb',\n",
       " 'cameo',\n",
       " 'sisu',\n",
       " 'mzspec',\n",
       " 'crisprdb',\n",
       " 'gnpis',\n",
       " 'licebase',\n",
       " 'sugarbind',\n",
       " 'rnacentral',\n",
       " 'microscope',\n",
       " 'swissregulon',\n",
       " 'validatordb',\n",
       " 'biominder',\n",
       " 'neurovault.collection',\n",
       " 'vmhmetabolite',\n",
       " 'mobidb',\n",
       " 'neurovault.image',\n",
       " 'nmr',\n",
       " 'vmhreaction',\n",
       " 'ascl',\n",
       " 'fb',\n",
       " 'gudmap',\n",
       " 'eu89h']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prefixes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are #629 prefixes registered in identifiers.org\n"
     ]
    }
   ],
   "source": [
    "print(\"There are #{} prefixes registered in identifiers.org\".format(len(prefixes)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
